<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Analysis Power Tools: MANUAL: apt-summary-genotype-axiom (2.12.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt-summary-genotype-axiom (2.12.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
<li><a href="#references">References</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>apt-summary-genotype-axiom</b> is a program that enables genotyping using a summary.a5 or .txt file as input, rather than CEL files</p>
<p> The basic requirements for running apt-summary-genotype-axiom:
<ul>
<li>Summary.txt or summary.a5 file</li>
<li>SNP priors model file</li>
</ul>
</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<pre class="fragment">
apt-summary-genotype-axiom \
--summary-a5-input-file $INDIR/AxiomGT1.summary.a5 \
--gender-file $INDIR/AxiomGT1.report.txt \
--analysis-files-path $LIBDIR \
--arg-file $LIBDIR/Axiom_PMDA_96orMore_all_cdf.r7.apt-summary-genotype-axiom.mm.SnpSpecificPriors.AxiomGT1.apt2.xml \
--out-dir $OUTDIR \
--log-file $OUTDIR/apt-genotype-axiom.log \
--batch-folder $OUTDIR \
--genotyping-node:snp-posteriors-output true
</pre>

or, if artifact-reduction-trustcheck-file is available:

<pre class="fragment">
apt-summary-genotype-axiom \
--summary-a5-input-file $INDIR/AxiomGT1.summary.a5 \
--gender-file $INDIR/AxiomGT1.report.txt \
--artifact-reduction-trustcheck-file  $INDIR/AxiomGT1.trustcheck.txt \
--analysis-files-path $LIBDIR \
--arg-file $LIBDIR/Axiom_PMDA_96orMore_all_cdf.r7.apt-summary-genotype-axiom.mm.SnpSpecificPriors.AxiomGT1.apt2.xml \
--out-dir $OUTDIR \
--log-file $OUTDIR/apt-genotype-axiom.log \
--batch-folder $OUTDIR \
--genotyping-node:snp-posteriors-output true
</pre>

<h2>Options: </h2>
<pre class="fragment">
   'apt-summary-genotype-axiom.exe'
   v2.12.0-rc4 https://github.com/thermofisher/ma-alg-apt2-genotyping f558778(80ecaa3) 2024-02-06
	
USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-s
elect WARNING,summary


  --arg-file (filename) = Read arguments from this file.    File should be a valid XML file, and should contain elements of the form
    <Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" />.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --out-dir (string8) = The output directory for result files.

  --analysis-files-path (dirname) = Default directory to search for analysis library files

  --batch-folder (string8) = Full path to Agatha batch folder. All genotyping data will be streamed into this folder as'all_genotypes_by_snps.CHP.bin' and 'all_genotypes_by_snps.CHP.index.txt'.

  --batch-folder-data-dir (string8) = Name of directory inside batch-folder where the analysis suite data is written.
    default value: 'AxiomAnalysisSuiteData'

  --analysis-name (string8) = Name to prepend to output files.

  --report (bool) = Print report.txt file of summary and genotyping statistics.
    default value: 'True'

  --report-file (string8) = Optional full path of report.txt file.

  --summary-input-file (filename) = The summary input file.

  --summary-a5-input-file (filename) = The summary input file.

  --genotyping-node:snp-priors-input-file (string8,required) = The string that identifies the snp priors to use when genotyping.

  --zw-gender (bool) = If array is for avian species with zw-gender designation.
    default value: 'False'

  --probabilities-output (bool) = Output a file with comma-separated probabilities for BB,AB,AA,Ocean for each probeset and sample [default 'false']
    default value: 'False'

  --process-multi-alleles (bool) = Augment genotyping workflow to accomodate multi-allelic snps (allele count > 2)
    default value: 'False'

  --probeset-ids (string8) = Tab delimited file with column 'probeset_id' specifying probesets to genotype.

  --special-snps (string8) = File specifying non-autosomal chromosome information.

  --snp-specific-param-file (string8) = File specifying SNP-specific parameters overriding standard parameters in genotyping.
    File must have columns: ProbeSetName, ShellBarrier, Ocean, Confidence, CSepPen and CSepThr.
    Aliases: --ssp-file

  --gender-file (string8) = File specifying genders for each CEL file.    File must have columns cel_files and gender.
    Aliases: --read-genders

  --inbred-file (string8) = File specifying inbred het penalty for each CEL file
.
    File must have columns cel_files and inbred_het_penalty.    Aliases: --read-inbred

  --hints-file (string8) = File specifying seed genotypes instead of DM algorithm    File must have first column 'probeset_id' and one additional column per CEL file, with file basename as column name.   Rows must have probeset id for first column and integers corresponding to calls for remaining columns.    Calls represented by 'No Call' = -1, 'AA' = 0, 'AB' = 1, 'BB' = 2.
    Aliases: --genotypes

  --global-inbred-het-penalty (float64) = Specifying global inbred het penalty value for each CEL file.
    default value: '0.000000'
    Aliases: --global-inbred

  --blob-format-version (float32) = Blob format version for storing genotyping result. 1.0=legacy, 2.0=multiallelic/copynumber.
    default value: '2.000000'

  --sample-multiplier (int32) = Used for generating fake blob with large number of samples. Ex: 10 samples x 500 multiplier = 5000 fake samples.
    default value: '1'

  --use-copynumber-call-codes (bool) = Use copynumber aware call codes, instead of raw brlmmp call codes.
    default value: 'False'

  --artifact-reduction-trustcheck-file (filename) = The trustcheck file produced  y artifact-reduction.

  --genotyping-node:snp-priors-input-file (string8,required) = The string that identifies the snp priors to use when genotyping.

  --genotyping-node:snp-posteriors-output (bool) = Write snp posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-models

  --genotyping-node:snp-posteriors-output-file (string8) = Optional full path of snp-posteriors.txt file.

  --genotyping-node:table-output (bool) = Output matching matrices of tab delimited genotype calls and confidences. [default 'true']
    default value: 'True'

  --genotyping-node:calls-file (string8) = Optional full path of calls.txt file.

  --genotyping-node:confidences-file (string8) = Optional full path of confidences.txt file.

  --genotyping-node:probabilities-file (string8) = Optional full path of probabilities template file.  The file is typically partitioned and integers inserted into the filename before the .txt suffix.

  --genotyping-node:probabilities-file-sample-count (int32) = Number of samples per probability file.  These files can be large, especially when running 1000's of samples. [default '0' = don't break up the file]
    default value: '0'
    Aliases: --prob-file-sample-count

  --genotyping-node:copynumber-probeset-calls-file (string8) = Optional full path of probeset copynumber calls file.

  --genotyping-node:copynumber-override-special-snps (bool) = Replace specialSnp copynumber gender designation with copynumber call
    default value: 'False'

  --genotyping-node:brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --genotyping-node:brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --genotyping-node:brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '1'

  --genotyping-node:brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --genotyping-node:brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --genotyping-node:brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --genotyping-node:brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --genotyping-node:brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --genotyping-node:brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --genotyping-node:brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --genotyping-node:brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --genotyping-node:brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --genotyping-node:brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --genotyping-node:brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --genotyping-node:brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --genotyping-node:brlmmp-transform (string8) = brlmmp transform. (mva, rvt, ssf, ces, assf, ccs)
    default value: 'mva'

  --genotyping-node:brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --genotyping-node:brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --genotyping-node:brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '16.000000'

  --genotyping-node:brlmmp-comvar (bool) = boolean to indicate if common variance is used in all clusters.
    default value: 'True'

  --genotyping-node:brlmmp-K (float64) = Scaling Parameter to use in transformations.
    default value: '1.000000'

  --multi-allele-background-node:snp-priors-input-file (string8,required) = The string that identifies the snp priors to use when genotyping.

  --multi-allele-background-node:snp-posteriors-output (bool) = Write snp posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-models

  --multi-allele-background-node:snp-posteriors-output-file (string8) = Optional full path of snp-posteriors.txt file.

  --multi-allele-background-node:table-output (bool) = Output matching matrices of tab delimited genotype calls and confidences. [default 'true']
    default value: 'True'

  --multi-allele-background-node:calls-file (string8) = Optional full path of calls.txt file.

  --multi-allele-background-node:confidences-file (string8) = Optional full path of confidences.txt file.

  --multi-allele-background-node:copynumber-probeset-calls-file (string8) = Optional full path of probeset copynumber calls file.

  --multi-allele-background-node:copynumber-override-special-snps (bool) = Replace specialSnp copynumber gender designation with copynumber call
    default value: 'False'

  --multi-allele-background-node:brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --multi-allele-background-node:brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --multi-allele-background-node:brlmmp-CM (int32) = brlmmp callmethod (1=batchmode, 2=single sample mode)
    default value: '1'

  --multi-allele-background-node:brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --multi-allele-background-node:brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --multi-allele-background-node:brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --multi-allele-background-node:brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --multi-allele-background-node:brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --multi-allele-background-node:brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --multi-allele-background-node:brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-copytype (int32) = brlmmp copytype. 0 =standard copy qc method
    default value: '0'

  --multi-allele-background-node:brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --multi-allele-background-node:brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --multi-allele-background-node:brlmmp-transform (string8) = brlmmp transform.(mva, rvt, ssf, ces, assf, ccs)
    default value: 'mva'

  --multi-allele-background-node:brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '16.000000'

  --multi-allele-background-node:brlmmp-comvar (bool) = boolean to indicate if common variance is used in all clusters.
    default value: 'True'

  --multi-allele-background-node:brlmmp-K (float64) = Scaling Parameter to use in transformations.
    default value: '1.000000'

  --multi-allele-pairwise-node:snp-priors-input-file (string8,required) = The string that identifies the snp priors to use when genotyping.

  --multi-allele-pairwise-node:snp-posteriors-output (bool) = Write snp posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-models

  --multi-allele-pairwise-node:snp-posteriors-output-file (string8) = Optional full path of snp-posteriors.txt file.

  --multi-allele-pairwise-node:table-output (bool) = Output matching matrices of tab delimited genotype calls and confidences. [default 'true']
    default value: 'True'

  --multi-allele-pairwise-node:calls-file (string8) = Optional full path of calls.txt file.

  --multi-allele-pairwise-node:confidences-file (string8) = Optional full path of confidences.txt file.

  --multi-allele-pairwise-node:copynumber-probeset-calls-file (string8) = Optional full path of probeset copynumber calls file.

  --multi-allele-pairwise-node:copynumber-override-special-snps (bool) = Replace specialSnp copynumber gender designation with copynumber call
    default value: 'False'

  --multi-allele-pairwise-node:brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --multi-allele-pairwise-node:brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --multi-allele-pairwise-node:brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '1'

  --multi-allele-pairwise-node:brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --multi-allele-pairwise-node:brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --multi-allele-pairwise-node:brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --multi-allele-pairwise-node:brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --multi-allele-pairwise-node:brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --multi-allele-pairwise-node:brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --multi-allele-pairwise-node:brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --multi-allele-pairwise-node:brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --multi-allele-pairwise-node:brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-clustertype (int32) = brlmmp clustertype.1 = standard 1-d clustering only
    default value: '1'

  --multi-allele-pairwise-node:brlmmp-transform (string8) = brlmmp transform. (mva, rvt, ssf, ces, assf, ccs)
    default value: 'mva'

  --multi-allele-pairwise-node:brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-inflatePRA (float64) = brlmmp inflatePRA.0 = no increase in uncertainty
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '16.000000'

  --multi-allele-pairwise-node:brlmmp-comvar (bool) = boolean to indicate if common variance is used in all clusters.
    default value: 'True'

  --multi-allele-pairwise-node:brlmmp-K (float64) = Scaling Parameter to use in transformations.
    default value: '1.000000'

  --multi-freq-flag (bool) = Adjustment factor for the frequency of the cluster.
    default value: 'False'

  --multi-inflate-PRA (float64) = Inflate amount of the posteriors covariance matrices
    default value: '0.000000'

  --multi-ocean (float64) = Analogous to the brlmmp ocean parameter but used in multi-allele genotyping.
    default value: '0.000010'

  --multi-lambda-P (float64) = Coefficient for pooling common variance.
    default value: '0.000000'

  --multi-wobble (float64) =
    default value: '0.050000'

  --multi-copy-distance-0to1 (float64) = Used to adjust empty clusters positions based on the nonempty cluster data
    default value: '1.500000'

  --multi-copy-distance-1to2 (float64) = Used to adjust empty clusters positions based on the nonempty cluster data
    default value: '0.200000'

  --multi-shell-barrier-0to1 (float64) = Push copy number 0 clusters away from copy number 1 clusters.
    default value: '0.050000'

  --multi-shell-barrier-1to2 (float64) = Push the copy number 2 cluster away from copy number 1 clusters.
    default value: '0.050000'

  --multi-confidence-threshold (float64) = Assign NoCall if the genotyping confidence is larger than this number.
    default value: '0.150000'

  --multi-priors-input-file (string8,required) = The string that identifies the multi-allelic priors to use when genotyping.

  --multi-posteriors-output (bool) = Write multi-allelic posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-multi-models

  --multi-posteriors-output-file (string8) = Full path of multi-allelic posteriors file.

  --@help       = print developer help



</pre><p><a class="anchor" id="faq"></a> </p>
<h2>Frequently Asked Questions </h2>
<p><a class="anchor" id="references"></a> </p>
<h2>References </h2>
<ul>
<li>BRLMM-P - <a href="http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf">http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf</a></li>
<li>PLIER - <a href="http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf">http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf</a> </li>
</ul>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Monday March 11 2024 12:27:00 for Analysis Power Tools by&nbsp;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
