<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Analysis Power Tools: MANUAL: apt-format-result (2.12.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt-format-result (2.12.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>apt-format-result</b> is an application which allows for the creation of VCF, or PLINK  file formats as well as IGV compatible files from Axiom analyzed batches.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<p>For each output type, <b>apt-format-result</b> requires an annotation file and either a <code>--calls-file</code> parameter or a <code>--batch-folder</code> parameter as input.</p>

<h3>Creating Text format files</h3>
<p>To create Text format files use the commands below: </p>
<pre class="fragment">    apt-format-result 
	--batch-folder &lt;folder location&gt;  
	--annotation-file &lt;file&gt; 
	--export-txt-file  &lt;file&gt; </pre>

<h3>Creating VCF format files</h3>
<p>To create VCF format files use the commands below: </p>
<pre class="fragment">    apt-format-result 
	--calls-file &lt;file&gt; 
	--annotation-file &lt;file&gt; 
	--export-vcf-file  &lt;file&gt
</pre><h3>Creating PLINK format files</h3>
<p>To create standard PLINK format (sample-based by rows) files use one of the following commands: </p>

<pre class="fragment">    apt-format-result 
	--calls-file &lt;file&gt; 
	--annotation-file &lt;file&gt; 
	--export-plink-file &lt;file&gt
</pre><p>To create transposed PLINK format (transposed is snp-based by rows) files use the following commands: </p>
<pre class="fragment">    apt-format-result 
	--calls-file &lt;file&gt; 
	--annotation-file &lt;file&gt; 
	--export-plinkt-file
</pre><h3>Creating IGV Compatible files</h3>
<pre class="fragment"> apt-format-result
--batch-folder &ltbatch folder&gt
--annotation-file &ltfile&gt
--run-format-igv True
--igv-cndata-file &ltcnv.a5 file&gt
--igv-output-prefix &ltfile name&gt
--igv-export-cnv-baf Default False
--igv-export-cnv-log2ratio Default False
--igv-export-seg-cn Default False
--igv-export-cnv-copynumber Default False
--igv-export-cnv-smooth-signal Default False
--igv-export-seg-loh Default False
--igv-out-dir &ltoutput dir location name&gt
</pre><h3>Additional Options</h3>
<p>The following optional commands may be added to the command string as desired:</p>
<ul>
<li><em>snp-list-file</em>: filters output to only the probesets of interest. It is highly recommended to supply this parameter with the Recommended.ps file created by ps-classification engine, which includes only the BestandRecommended=1 probesets from the ps.performance.txt file.</li>
<li><em>snp-identifier-column</em>: allows a column name to be used as SNP identifier for export. This must be one of annotation columns, e.g. {Affy_SNP_ID, dbSNP_RS_ID}. The default value for this is 'probeset_id'</li>
<li><em>pedigree-file</em>: this file can be used if you wish to include basic pedigree information such as family, sample, maternal, ect.</li>
<li><em>export-call-format</em>: sets format of export calls. Available formats: call_code {AA/AB/BB} or base_call {CC/CT/TT} or translated {0/1/2/-1}. Default is call_code. â€ƒ</li>
</ul>
<p><a class="anchor" id="options"></a> </p>
<h2>Options</h2>
<pre class="fragment">
   'apt-format-result.exe'
   v2.12.0-rc2 https://github.com/thermofisher/ma-alg-apt2-format-result c001a73(033b679) 2023-10-04


Usage:

  apt-format-result --batch-folder <folder> --annotation-file <file> [options]

  or

  apt-format-result --calls-file <file> --annotation-file <file> [options]

  or

  apt-format-result --input-plink-file <file> --export-txt-file <txt-file> --export-vcf-file <vcf-file> [options]

  or

  apt-format-result [--run-format-cnv true] [cnv-options]

  or

  apt-format-result [--run-format-igv true] [igv-options]


USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    <Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" />.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --run-format-cnv (bool) = Export CNV data in Nexus Copy Number format.
    default value: 'False'

  --run-format-igv (bool) = Export CNV data in IGV format.
    default value: 'False'

  --batch-folder (dirname) = Full path for Agatha batch folder that contains genotyping data.

  --batch-folder-data-dir (string8) = Name of directory inside batch-folder where the analysis suite data is written.
    default value: 'AxiomAnalysisSuiteData'

  --calls-file (filename) = CallsFile.txt file that was generated by apt2 genotyping software.

  --sample-filter-file (filename) = List of sample names to filter for export. Generated by user or AxAS. Samples must exist in batch folder. Single column with header of 'cel_files' and sample names only, no paths.

  --sample-attributes-file (filename) = Sample attributes file in IGV format. Generated by user or AxAS. Samples must exist in batch folder. Expected two header columns: [Sample Filename] and [Alternate Sample Name]

  --export-alternate-sample-names (bool) = Must be use in conjunction with --sample-attributes-file that contains [Alternate Sample Name] column name.
    default value: 'False'

  --annotation-file (string8) = Full path for annotation file, e.g. sqlite db.

  --annotation-columns (string8) = List of comma-delimited column names to be extracted from annotation file. Ex: 'rs_id,chromosome,position'.

  --snp-list-file (filename) = Full path of file containing list of selected SNP's for export.

  --snp-identifier-column (string8) = Column name to be used as snp identifier for export. Must be one of annotation columns, e.g. {Affy_SNP_ID, dbSNP_RS_ID}.
    default value: 'probeset_id'

  --pedigree-file (filename) = Contains basic pedigree information such as identifiers for family, sample, paternal, maternal and etc. Data comes from GUI.

  --cn-region-calls-file (filename) = Full path of file containing list of copy number states in each chromosome region for VCF export.

  --cn-region-calls-use-cn-raw (bool) = If True, report CN_raw from cn-region-calls-file.  If False, report CN_State.
    default value: 'False'

  --export-confidence (bool) = Turn on/off exporting of confidence data
    default value: 'False'

  --export-log-ratio (bool) = Turn on/off exporting of log2 ratio (MvA) and contrast (CES)
    default value: 'False'
    Aliases: --export-contrast

  --export-strength (bool) = Turn on/off exporting of strength (MvA) and size (CES)
    default value: 'False'
    Aliases: --export-size

  --export-allele-signals (bool) = Turn on/off exporting of strength data
    default value: 'False'

  --export-chr-shortname (bool) = Use chromosome shortname in annotation sqlite db if available.
    default value: 'False'

  --export-single-samples (bool) = One sample per file
    default value: 'False'

  --export-call-format (string8) = Sets format of export calls. Available formats: call_code {AA/AB/BB} or base_call {CC/CT/TT} or translated {0/1/2/-1}. Default is call_code.
    default value: 'call_code'

  --export-basecall-otv (string8) = Use in conjunction with '--export-call-format base_call' to make a distinction between NoCall and OTV when exporting using AxiomGT1.calls.txt. Both will be replaced with '---' per DIR specification if this parameter is not supplied. (ex: OTV)

  --export-alternate-probeset-ids (string8) = File with alternate probeset ids

  --@export-plink-multiallele-snps (bool) = Whether to export multiallele snps for PLINK export.
    default value: 'False'

  --enable-divider (bool) = Enable divider '/' between alleles
    default value: 'True'

  --filter-pass-probeset (bool) = Filter pass probesets in vcf file
    default value: 'False'

  --input-plink-file (filename) = Full path for PLINK file.

  --plink-allele-delimited (string8) = Sets the delimiter between allele values during reconverting tPLINK to VCF. default space.
    default value: ' '

  --plink-columns-delimited (string8) = Sets the delimiter between columns during reconverting tPLINK to VCF. default tab.
    default value: '	'

  --plink-abstract-callcodes (bool) = Export plink format using abstract call codes, e.g. {AA, AB, BB}.
    default value: 'False'

  --plink-missing-genotype (string8) = Use specified character as missing genotype for plink export.
    default value: '0'

  --plink-sort-by-chrpos (bool) = Sort PLINK .map and .tped files by chromosome and position, in .ped file the genotype calls should be in the order of chr/pos from left to right, based on .map file.
    default value: 'False'

  --performance-file (filename) = Full path for performance file.

  --ps-performance-qc (vec_string8) = Additional fields in the performance file to put to the INFO column of VCF file

  --additional-snp-information-file (filename) = TSV file with additional SNP information.

  --exclude-unknown-chr (bool) = When true, output will only contain calls from standard chromosomes which are well characterized per species
    default value: 'False'

  --export-txt-file (string8) = Export results to TXT format.

  --export-plink-file (string8) = Export results to standard plink format, e.g {MAP, PED}. (standard is sample-based by rows)

  --export-plinkt-file (string8) = Export results to transposed plink format, e.g. {tFAM, tMAP, tPED}. (transposed is snp-based by rows)

  --export-vcf-file (string8) = Export results to VCF format.

  --cnv-cndata-file (filename,required) = AxiomCN.cnv.a5 data file.

  --cnv-sample-filter-file (filename) = File with the list of samples to export. Samples must be in current batch. Single column with header of 'cel_files' and sample names only, no paths.

  --cnv-snp-list-file (filename) = Full path of file containing list of selected SNP's for export.

  --cnv-sample-attributes-file (filename) = Sample attributes file in IGV format. Generated by user or AxAS. Expected two header columns: [Sample Filename] and [Alternate Sample Name]

  --cnv-export-alternate-sample-names (bool) = Must be used in conjunction with --cnv-sample-attributes-file that contains [Alternate Sample Name] column name.
    default value: 'False'

  --cnv-out-dir (string8,required) = Export output folder path. Relative/absolute supported. If path does not exist, an attempt is made to create it.

  --igv-cndata-file (filename,required) = AxiomCN.cnv.a5 data file.

  --igv-sample-filter-file (filename) = File with the list of samples to export. Samples must be in current batch. Single column with header of 'cel_files' and sample names only, no paths.

  --igv-sample-attributes-file (filename) = Sample attributes file in IGV format. Generated by user or AxAS. Expected two header columns: [Sample Filename] and [Alternate Sample Name]

  --igv-snp-list-file (filename) = Full path of file containing list of selected SNP's for export.

  --igv-export-seg-cn (bool) = Export copynumber state segment information
    default value: 'False'

  --igv-export-seg-loh (bool) = Export loss of heterozygosity (LOH) segment information
    default value: 'False'

  --igv-export-cnv-baf (bool) = Export CNV B-allele frequency (BAF) information
    default value: 'False'

  --igv-export-cnv-copynumber (bool) = Export CNV copynumber information
    default value: 'False'

  --igv-export-cnv-log2ratio (bool) = Export CNV log2ratio information
    default value: 'False'

  --igv-export-cnv-smooth-signal (bool) = Export CNV smooth signal information
    default value: 'False'

  --igv-export-sample-attributes (bool) = Must be used in conjunction with --igv-sample-attributes-file that contains [Alternate Sample Name] column name.
    default value: 'False'

  --igv-export-alternate-sample-names (bool) = Must be used in conjunction with --igv-sample-attributes-file that contains [Alternate Sample Name] column name.
    default value: 'False'

  --igv-out-dir (dirname,required) = Export output folder path. Relative/absolute supported. If path does not exist, an attempt is made to create it.

  --igv-output-prefix (string8) = Output prefix append at the start of output file names. Current timestamp will be used if left empty.

  --@help       = print developer help

</pre> </div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Monday March 11 2024 12:27:00 for Analysis Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
