<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Analysis Power Tools: MANUAL: apt-geno-qc-axiom (2.11.8)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt-geno-qc-axiom (2.11.8) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction</h2>
<p><b>apt-geno-qc-axiom</b> is a single-sample quality control tool for genotyping arrays that reports metrics like <code>axiom_dishqc_DQC</code>.</p>
<p>For assessing Axiom&trade; arrays with this tools, see <a href="vignette-axiom-geno-qc-axiom.html">Axiom&trade; Single-Sample QC Analysis</a>.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2><a class="anchor" id="QuickStart"></a>
Quick Start</h2>
<p>An Axiom&trade; example: </p>
<pre class="fragment">
  apt-geno-qc-axiom \
    --analysis-files-path /library/file/path \
    --arg-file Axiom_GW_Hu_SNP.r4.apt-geno-qc.AxiomQC1.xml \
    --cel-files cel_file_list.txt \
    --out-file Geno-qcResults.txt
</pre>
<p>A SNP6 example: </p>
<pre class="fragment">
apt-geno-qc-axiom \
  --cdf-file GenomeWideSNP_6.cdf \
  --qca-file GenomeWideSNP_6.r2.qca \
  --qcc-file GenomeWideSNP_6.r2.qcc \
  --chrX-probes GenomeWideSNP_6.chrXprobes \
  --chrY-probes GenomeWideSNP_6.chrYprobes \
  --cel-files cel_file_list.txt \
  --out-file results.txt
</pre>
<p><a class="anchor" id="options"></a></p>
<h2>Options</h2>
<pre class="fragment">
   'apt-geno-qc-axiom.exe'
   APT_2.11.8-rc3 https://github.com/thermofisher/ma-alg-apt2-genotyping c84680a(a9a90c6) 2023-02-03

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    <Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" />.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --max-threads (int32) = Maximum thread count (0 = unlimited threads); Default = number of CPUs detected
    default value: '0'
    Aliases: --mt, --max_threads

  --out-file (filename) = Name to use for the output file.

  --out-dir (dirname) = The output directory for result files.

  --temp-dir (string8) = The output directory for temporary files.

  --analysis-files-path (dirname) = Default directory to search for analysis library files

  --chip-type (vec_string8) = Text identifier for chip.
    default value: 'apt2-genotype'
    Aliases: --array-type

  --force (bool) = Force the files to process regardless of array type.
    default value: 'True'

  --analysis-name (string8) = Name to prepend to output files.
    default value: 'apt-geno-qc-axiom'

  --x-probes-file (filename) = The file that identifies the X chromosome probe ids.
    Aliases: --chrX-probes

  --y-probes-file (filename) = The file that identifies the Y chromosome probe ids.
    Aliases: --chrY-probes

  --w-probes-file (filename) = The file that identifies the W chromosome probe ids.
    Aliases: --chrW-probes

  --z-probes-file (filename) = The file that identifies the Z chromosome probe ids.
    Aliases: --chrZ-probes

  --sketch-target-input-file (string8) = Name of the file containing sketch to use.
    Aliases: --target-sketch

  --sketch-target-scale-value (float64) = Target to scale the sketch to
    default value: '1000.000000'

  --sketch-target-use-avg (bool) = When there is a range of identical intensities, use the average of the corresponding target sketch values.
    default value: 'True'

  --dual-channel-target-sketch (bool) = Create sketches from separate channels (AT/GC)
    default value: 'True'

  --dual-channel-normalization (bool) = Use dual channel normalization.
    default value: 'True'

  --qca-file (string8,required) = File defining QC analysis methods.

  --qcc-file (string8,required) = File defining QC probesets.

  --reagent-kit-discriminator (string8) = File containing list of probesets with PC1s and means to use for classifying the reagent kits.

  --cdf-file (string8) = The CDF input file.

  --cel-files (string8,required) = The file that identifies the cel files to process.

  --probe-class-file (string8) = File containing probe_id (1-based) of probes and a 'class' designation. Used to compute mean probe intensity by class for report file.

  --sketch-size (int32,required) = The sketch size to use.

  --igender-female-threshold (float64) = The female threshold to use in the raw intensity gender call.
    default value: '0.480000'
    Aliases: --female-thresh

  --igender-male-threshold (float64) = The male threshold to use in the raw intensity gender call.
    default value: '0.700000'
    Aliases: --male-thresh

  --@help       = print developer help
</pre><p><a class="anchor" id="faq"></a> </p>
<h2><a class="anchor" id="manFaq"></a>
Frequently Asked Questions</h2>
<p>Q. What is a probe_id?</p>
<p>A. See the <a href="FAQ.html#probe_id_faq">FAQ item on probe IDs</a> for more info. </p>
</div>
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Friday February 24 2022 11:11:00 for Analysis Power Tools by&nbsp;
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
