<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Analysis Power Tools: MANUAL: apt-copynumber-axiom-hmm (2.12.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>MANUAL:  apt-copynumber-axiom-hmm (2.12.0) </h1>  </div>
</div>
<div class="contents">
<h2><a class="anchor" id="Contents"></a>
Contents</h2>
<ul>
<li>
<a href="#intro">Introduction.</a> </li>
<li>
<a href="#quickstart">Quick Start</a> </li>
<li>
<a href="#options">Options</a> </li>

</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p><b>apt-copynumber-axiom-hmm</b> implements copy number analysis
for the Axiom family of arrays. The program reports copy number and loss of
heterozygosity (LOH) states using a hidden Markov model.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2><a class="anchor" id="QuickStart"></a>
Quick Start</h2>
<p>Basic usage example:</p>
<div class="fragment"><pre class="fragment">
apt-copynumber-axiom-hmm \
--analysis-files-path &lt;library_directory&gt; \
--arg-file &lt;library_directory&gt;/arrayname.r1.apt-copynumber-axiom-cnvmix.AxiomHMM.apt2.xml \
--reference-file &lt;library_directory&gt;/arrayname.r1.cn_models \
--summary-file AxiomGT1.summary.a5 \
--report-file AxiomGT1.report.txt \
--out-dir &lt;out_directory&gt; \
--log-file &lt;out_directory&gt;/apt-copynumber-axiom-hmm.log
</div>
<p>The following example includes Loss of Heterozygosity detection. It also includes a custom hmm-regions.txt file to restrict segment calling to specified genomic regions.  The reference-file and regions-file parameters must be specified after the arg-file parameter, since the arg-file includes default settings for these parameters. The following example also uses a text version of the summary-file.  Note that the summary.txt format is slower than the .a5 format, especially for larger datasets. The summary.txt file can be converted to a5 format by using apt2-a5-table-converter.</p>
<div class="fragment"><pre class="fragment">
apt-copynumber-axiom-hmm \
--analysis-files-path &lt;library_directory&gt; \
--arg-file &lt;library_directory&gt;/arrayname.r1.apt-copynumber-axiom-cnvmix.AxiomHMM.apt2.xml \
--reference-file &lt;library_directory&gt;/arrayname.r1.cn_models \
--regions-file &lt;library_directory&gt;/custom.hmm_regions.txt \
--use-text-format-summary-file TRUE \
--summary-file AxiomGT1.summary.txt \
--report-file AxiomGT1.report.txt \
--loh-calls-file AxiomGT1.calls.txt \
--loh-confidences-file AxiomGT1.confidences.txt \
--out-dir &lt;out_directory&gt; \
--log-file &lt;out_directory&gt;/apt-copynumber-axiom-hmm.log
</div>
<p>Segment filtering is on by default. Refer to parameters beginning with --seg-min for more information.</p>
<p>Refer to the Axiom Copy Number Data Analysis Guide for more information, like how to reformat the outputs as needed.</p>
<p><a class="anchor" id="options"></a> </p>
<h2><a class="anchor" id="options"></a>
<h2>Options</h2>
<div class="fragment"><pre class="fragment">
    'apt-copynumber-axiom-hmm.exe'
   v2.12.0-rc3 https://github.com/thermofisher/ma-alg-apt2-genotyping 666978e(80ecaa3) 2024-01-04

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form

    &lt;Parameter name=&quot;PARAMETER-NAME&quot;
    analysis=&quot;NODE-NAME&quot; currentValue=&quot;VALUE&quot; /&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --delete-temp-files (bool) = Delete the temporary files after the run completes.
    default value: 'True'

  --wave-count (int32) = Wave correction wave count.
    default value: '-1'

  --out-dir (string8) = Output directory for result files

  --analysis-name (string8) = Name to prepend to output files
    default value: 'AxiomHMM'

  --analysis-files-path (string8) = Default directory to search for analysis library files

  --summary-file (filename) = Summary file generated by apt-genotype-axiom (used as input)

  --reference-file (filename) = Reference file

  --mapdc-max (float64) = Samples with MAPDc greater than this threshold will fail quality control. MAPDc may be computed if control samples are used for plate correction
    default value: '0.230000'

  --waviness-sdc-max (float64) = Samples with WavinessSDc greater than this threshold will fail quality control. WavinessSDc may be computed if control samples are used for plate correction
    default value: '0.100000'

  --mapd-max (float64) = Samples with MAPD greater than this threshold will fail quality control if MAPDc is not reported
    default value: '0.400000'

  --waviness-sd-max (float64) = Samples with WavinessSD greater than this threshold will fail quality control if WavinessSDc is not reported
    default value: '0.300000'

  --chip-type (vec_string8) = Text identifier for chip.
    Aliases: --array-type

  --use-wave-correction (bool) = Use wave correction
    default value: 'False'

  --calculate-allelic-differences (bool) = Allelic differences will be added to the output
    default value: 'False'

  --use-baf-acr (bool) = Use BAF-based ACR
    default value: 'False'

  --use-ad-acr (bool) = Use AD-based ACR
    default value: 'False'

  --intensity-covariate-flags (filename) = A key-value covariate-flag text file overriding the hardcoded mapping of intensity covariate names to process flags

  --lr-covariate-flags (filename) = A key-value covariate-flag text file overriding the hardcoded mapping of log2 ratio covariate names to process flags

  --use-text-format-summary-file (bool) = Use Axiom summary file in text format instead of a5
    default value: 'False'

  --exclude-failed-samples (bool) = Include samples that failed QC in the output
    default value: 'False'

  --compute-plate-corrected-CN-QC (bool) = If true and control samples are used for plate correction, MAPDc and WavinessSDc are computed and the CN QC status of samples will be based on these metrics
    default value: 'False'

  --human-X-chromosome (string8) = Defines the X chromosome name
    default value: 'X'

  --human-Y-chromosome (string8) = Defines the Y chromosome name
    default value: 'Y'

  --cn-regions-file (filename) = File that lists the regions for the CN calculation

  --report-file (filename) = Report file generated by apt-genotype-axiom (used as input)

  --controls-file (filename) = The file containing list of control sample file names used for per-plate signal correction

  --y-probes-file (filename) = The file that identifies the Y chromosome probe set names

  --regions-file (filename,required) = The file that defines regions and HMM parameters

  --samples-per-block (int32) = Number of samples per block to load from summary HDF5 file. Default is 200.
    default value: '200'

  --loh-batch-folder (filename) = The batch folder generated by apt-genotype-axiom (used as input). Set either the loh-batch-folder or loh-calls-file and loh-confidences-file to get LOH segments.

  --loh-calls-file (filename) = The calls file generated by apt-genotype-axiom (used as input)

  --loh-confidences-file (filename) = The confidences file generated by apt-genotype-axiom (used as input)

  --loh-samples-per-block (int32) = Number of samples per block to process from LOH batch folder or calls file. Only supports up to 200.
    default value: '200'

  --loh-error-rate (float64) = Error rate to use in LOH processing
    default value: '0.050000'

  --loh-beta (float64) = Beta to use in LOH processing
    default value: '0.001000'

  --loh-alpha (float64) = Alpha to use in LOH processing
    default value: '0.010000'

  --loh-separation (int32) = Separation to use in LOH processing
    default value: '1000000'

  --loh-min-marker-count (int32) = Minimum marker count to use in LOH processing
    default value: '10'

  --loh-no-call-threshold (float64) = No call threshold to use in LOH processing
    default value: '0.050000'

  --loh-min-genomic-span (int32) = Minimum genomic span to use in LOH processing
    default value: '1000000'

  --ad-outlier-trim (float64) = The trim value to use for alleleic difference outliers.
    default value: '3.000000'

  --nearest-power-of-two (int32) = Number of point to use in the FFT-based density fitting.
    default value: '512'

  --allele-peak-count (int32) = Maximum allowable number of peaks in Allelic Difference Modulation
    default value: '24'

  --ap-ad-step (int32) = Step to use in calculating allele peaks
    default value: '25'

  --ap-ad-window (int32) = Window to use in calculating allele peaks
    default value: '150'

  --ap-ad-point-count (int32) = Point count to use in calculating allele peaks
    default value: '129'

  --ap-ad-bandwidth (float64) = Bandwidth to use in calculating allele peaks
    default value: '0.300000'

  --ap-ad-cutoff (float64) = Cutoff to use in calculating allele peaks
    default value: '0.000000'

  --ap-ad-threshold (float64) = Clean threshold to use in calculating allele peaks
    default value: '0.300000'

  --ap-ad-height-threshold (float64) = Height threshold factor to use in calculating allele peaks
    default value: '3.000000'

  --ap-ad-height-threshold-bound (float64) = Bounds between which the height threshold is applied
    default value: '1.000000'

  --ap-ad-symmetry (bool) = Whether to symmetrize the data when calculating allele peaks
    default value: 'True'

  --ap-ad-shrink-factor-3 (float64) = shrink-factor-3
    default value: '0.500000'

  --ap-ad-shrink-factor-4 (float64) = shrink-factor-4
    default value: '0.600000'

  --ap-ad-fld-weights (filename) = Name of file with FLD-derived weights for allelic differences

  --snp-qc-snp-list (filename) = SNPs to use in calculating SNP QC.

  --wave-bandwidth (int32) = Wave correction bandwidth.
    default value: '101'

  --wave-bin-count (int32) = Wave correction bin count.
    default value: '25'

  --wave-smooth (bool) = Wave correction wave smooth.
    default value: 'False'

  --y-target (float64) = The Y chromosome target log2 ratio value.
    default value: '0.000000'

  --ap-baf-step (int32) = Step to use in calculating allele peaks
    default value: '25'
    Aliases: --baf-step

  --ap-baf-window (int32) = Window to use in calculating allele peaks
    default value: '150'
    Aliases: --baf-window

  --ap-baf-point-count (int32) = Point count to use in calculating allele peaks
    default value: '129'
    Aliases: --baf-point-count

  --ap-baf-bandwidth (float64) = Bandwidth to use in calculating allele peaks
    default value: '0.300000'
    Aliases: --baf-bandwidth

  --ap-baf-cutoff (float64) = Cutoff to use in calculating allele peaks
    default value: '0.000000'
    Aliases: --baf-cutoff

  --ap-baf-threshold (float64) = Clean threshold to use in calculating allele peaks
    default value: '0.300000'
    Aliases: --baf-threshold

  --ap-baf-height-threshold (float64) = Height threshold factor to use in calculating allele peaks
    default value: '3.000000'
    Aliases: --baf-height-threshold

  --ap-baf-height-threshold-bound (float64) = Bounds between which the height threshold is applied
    default value: '1.000000'
    Aliases: --baf-height-threshold-bound

  --ap-baf-symmetry (bool) = Whether to symmetrize the data when calculating allele peaks
    default value: 'True'
    Aliases: --baf-symmetry

  --ap-baf-shrink-factor-3 (float64) = ap-baf-shrink-factor-3
    default value: '0.500000'
    Aliases: --shrink-factor-3

  --ap-baf-shrink-factor-4 (float64) = ap-baf-shrink-factor-4
    default value: '0.600000'
    Aliases: --shrink-factor-4

  --ap-baf-fld-weights (filename) = Name of file with FLD-derived weights for BAFs

  --transform-data (bool) = Whether to transform the data
    default value: 'True'

  --mosaicism-alpha-cn-calibrate (float64) = 
    default value: '0.470000'

  --mosaicism-beta-cn-calibrate (float64) = 
    default value: '1.000000'

  --mosaicism-alpha-X-cn-calibrate (float64) = 
    default value: '0.450000'

  --mosaicism-beta-X-cn-calibrate (float64) = 
    default value: '1.000000'

  --mosaicism-alpha-Y-cn-calibrate (float64) = 
    default value: '0.400000'

  --mosaicism-beta-Y-cn-calibrate (float64) = 
    default value: '1.000000'

  --coverage-gap-threshold (int32) = Controls how to break up a chromosome into regions to be processed
    default value: '1000000'

  --prune-min-prob (float64) = Probability regulating how many run-lengths are kept
    default value: '0.000000'

  --hazard-lambda (float64) = Controls how likely it is that a new run will be started in the BOCPD
    default value: '800.000000'

  --min-pred-prob (float64) = Threshold value using for cutting off tails of posterior predictive probability distribution
    default value: '0.000000'

  --min-pred-prob-tail (bool) = When true, treat min-pred-prob as representing tail probability, when false treat min-pred-prob as representing pdf value
    default value: 'True'

  --hyper-mu-zero (float64) = The mean for the conjugate prior distribution in the BOCPD
    default value: '0.000000'

  --hyper-var-zero (float64) = The variance for the conjugate prior distribution in the BOCPD
    default value: '0.030000'

  --var-known (float64) = Specifies the assumed variance of the log2 ratios under a Gaussian likelihood model
    default value: '0.090000'

  --max-keep (int32) = Maximum number of active run lengths to keep (if negative, prune-min-prob is used to control pruning)
    default value: '50'

  --min-small-seg-size (int32) = Minimum segment size in the small segment remover
    default value: '20'

  --cutoff-neutral-upper (float64) = Upper bound of the neutral region for the joiner algorithm
    default value: '2.050000'

  --cutoff-neutral-lower (float64) = Lower bound of the neutral region for the joiner algorithm
    default value: '1.950000'

  --cutoff-non-neutral-upper (float64) = Upper bound of the non-neutral region for the joiner algorithm
    default value: '2.100000'

  --cutoff-non-neutral-lower (float64) = Lower bound of the non-neutral region for the joiner algorithm
    default value: '1.900000'

  --joiner-max-dist (float64) = Maximum difference in CCN between adjacent joinable segments
    default value: '0.100000'

  --joiner-use-cutoffs (bool) = Whether to group segments into separate classes (neutral/non-neutral, upper/lower)
    default value: 'True'

  --accept-genomic-threshold (int32) = Minimum genomic size allowable for a segment to be accepted as mosaic
    default value: '1000000'

  --accept-length-loss (string8) = A boundary control for the acceptance regions for loss mosaic calls on autosomes: regions lengths
    default value: '500,1000,1500,2000,5000'

  --accept-level-loss-lower (string8) = A boundary control for the acceptance regions for loss mosaic calls on autosomes: lower bounds
    default value: '1.25,1.25,1.2,1.2,1.2'

  --accept-level-loss-upper (string8) = A boundary control for the acceptance regions for loss mosaic calls on autosomes: upper bounds
    default value: '1.75,1.8,1.85,1.88,1.9'

  --accept-length-gain (string8) = A boundary control for the acceptance regions for gain mosaic calls on autosomes: regions lengths
    default value: '500,1000,1500,2000,5000'

  --accept-level-gain-lower (string8) = A boundary control for the acceptance regions for gain mosaic calls on autosomes: lower bounds
    default value: '2.28,2.2,2.18,2.15,2.1'

  --accept-level-gain-upper (string8) = A boundary control for the acceptance regions for gain mosaic calls on autosomes: upper bounds
    default value: '2.75,2.75,2.8,2.8,2.8'

  --accept-length-loss-X (string8) = A boundary control for the acceptance regions for loss mosaic calls on chromosome X: regions lengths
    default value: '500,1000,1500,2000,5000'

  --accept-level-loss-lower-X (string8) = A boundary control for the acceptance regions for loss mosaic calls on chromosome X: lower bounds
    default value: '1.25,1.25,1.2,1.2,1.2'

  --accept-level-loss-upper-X (string8) = A boundary control for the acceptance regions for loss mosaic calls on chromosome X: upper bounds
    default value: '1.75,1.8,1.85,1.88,1.9'

  --accept-length-gain-X (string8) = A boundary control for the acceptance regions for gain mosaic calls on chromosome X: regions lengths
    default value: '500,1000,1500,2000,5000'

  --accept-level-gain-lower-X (string8) = A boundary control for the acceptance regions for gain mosaic calls on chromosome X: lower bounds
    default value: '2.28,2.2,2.18,2.15,2.14'

  --accept-level-gain-upper-X (string8) = A boundary control for the acceptance regions for gain mosaic calls on chromosome X: upper bounds
    default value: '2.75,2.75,2.8,2.8,2.8'

  --accept-length-loss-Y (string8) = A boundary control for the acceptance regions for loss mosaic calls on chromosome Y: regions lengths
    default value: '500,1000,1500,2000'

  --accept-level-loss-lower-Y (string8) = A boundary control for the acceptance regions for loss mosaic calls on chromosome Y: lower bounds
    default value: '0.35,0.35,0.35,0.35'

  --accept-level-loss-upper-Y (string8) = A boundary control for the acceptance regions for loss mosaic calls on chromosome Y: upper bounds
    default value: '0.75,0.85,0.87,0.88'

  --accept-length-gain-Y (string8) = A boundary control for the acceptance regions for gain mosaic calls on chromosome Y: regions lengths
    default value: '500,1000,1500,2000'

  --accept-level-gain-lower-Y (string8) = A boundary control for the acceptance regions for gain mosaic calls on chromosome Y: lower bounds
    default value: '1.25,1.15,1.15,1.15'

  --accept-level-gain-upper-Y (string8) = A boundary control for the acceptance regions for gain mosaic calls on chromosome Y: upper bounds
    default value: '1.75,1.75,1.8,1.8'

  --multi-n-sep (int32) = Maximum separation between segments in a Type A multi-segment
    default value: '200'

  --multi-delta-max-A (float64) = Maximum difference in CCN between segments in a Type A multi-segment
    default value: '0.100000'

  --multi-delta-max-B (float64) = Maximum difference in CCN between segments in a Type B multi-segment
    default value: '0.400000'

  --multi-loss-boundaries (string8) = Maximal extent in CCN for loss regions on autosomes when applying multi-segment acceptance rules
    default value: '1.2,1.9'

  --multi-gain-boundaries (string8) = Maximal extent in CCN for gain regions on autosomes when applying multi-segment acceptance rules
    default value: '2.1,2.8'

  --multi-loss-boundaries-X (string8) = Maximal extent in CCN for loss regions on chromosome X when applying multi-segment acceptance rules
    default value: '1.2,1.9'

  --multi-gain-boundaries-X (string8) = Maximal extent in CCN for gain regions on chromosome X when applying multi-segment acceptance rules
    default value: '2.1,2.8'

  --multi-loss-boundaries-Y (string8) = Maximal extent in CCN for loss regions on chromosome Y when applying multi-segment acceptance rules
    default value: '0.3,0.9'

  --multi-gain-boundaries-Y (string8) = Maximal extent in CCN for gain regions on chromosome Y when applying multi-segment acceptance rules
    default value: '1.1,1.8'

  --mosaicism-covariates (filename) = File containing mosaicism preprocessing covariates

  --mosaicism-bin-count-autosomes-X (int32) = Number of bins to use when binning mosaicism covariate for autosomes and chrX
    default value: '100'

  --mosaicism-bin-count-Y (int32) = Number of bins to use when binning mosaicism covariate for chrY
    default value: '10'

  --mosaicism-binning-type (string8) = Mosaicism covariate binning method, either ‘discrete’ or ‘equally-populated
    default value: 'discrete'

  --special-snps-file (filename) = (unused) The special-snps file for male/female copy numbers

  --seg-min-bases-CN-zero (int32) = Minimum size in bases for segments that report copy number 0
    default value: '25000'

  --seg-min-bases-CN-one (int32) = Minimum size in bases for segments that report copy number 1
    default value: '50000'

  --seg-min-bases-CN-two (int32) = Minimum size in bases for segments that report copy number 2
    default value: '50000'

  --seg-min-bases-CN-three (int32) = Minimum size in bases for segments that report copy number 3
    default value: '100000'

  --seg-min-bases-CN-fourormore (int32) = Minimum size in bases for segments that report copy number greater than 3
    default value: '100000'

  --seg-min-bases-CN-oneormore (int32) = Minimum size in bases for segments that report copy number greater than 0 (Deprecated)
    default value: '0'

  --seg-min-probesets-CN-zero (int32) = Minimum size in probesets for segments that report copy number 0
    default value: '25'

  --seg-min-probesets-CN-one (int32) = Minimum size in probesets for segments that report copy number 1
    default value: '50'

  --seg-min-probesets-CN-two (int32) = Minimum size in probesets for segments that report copy number 2
    default value: '50'

  --seg-min-probesets-CN-three (int32) = Minimum size in probesets for segments that report copy number 3
    default value: '100'

  --seg-min-probesets-CN-fourormore (int32) = Minimum size in probesets for segments that report copy number 3
    default value: '100'

  --seg-min-probesets-CN-oneormore (int32) = Minimum size in probesets for segments that report copy number greater than 0 (Deprecated)
    default value: '0'

  --acr-min-exclude-proportion (float64) = Minimum fraction of BAF data points needed to trigger the ACR
    default value: '0.100000'

  --acr-lower-baf-bound (float64) = Lower threshold for BAF to be considered for the ACR
    default value: '0.400000'

  --acr-upper-baf-bound (float64) = Upper threshold for BAF to be considered for the ACR
    default value: '0.600000'

  --acr-min-segment-size (int32) = Minimum size for BAF-based ACR
    default value: '50'

  --acr-ad-means (string8) = Model means

  --acr-ad-stddevs (string8) = Model standard deviations

  --acr-ad-stddev_AB (float64) = Model standard deviation sigma_AB
    default value: '0.000000'

  --acr-ad-stddev_AABB (float64) = Model standard deviation sigma_AABB
    default value: '0.000000'

  --acr-ad-mixing (string8) = Model mixing parameters

  --acr-ad-min-segment-size (int32) = Minimum size for AD-based ACR
    default value: '50'

  --acr-ad-tuning (float64) = 
    default value: '1.000000'

  --acr-ad-reject-threshold (float64) = 
    default value: '0.000000'

  --max-folded-baf-hom (float64) = Folded BAF values between 0 and this parameter value will be assumed to represent homozygous genotypes and ignored
    default value: '0.150000'

  --min-folded-baf-markers (int32) = Median folded BAF values are calculated for a segment only if the number of markers in the segment with folded baf >= max-folded-baf-hom exceeds the value specified for this parameter
    default value: '10'

  --cn-gender-cutoff (float64) = CN Gender cutoff
    default value: '0.500000'

  --cn-state-threshold (float64) = CN Gender threshold for counting CN state as nonzero
    default value: '0.000000'

  --enforce-single-cn-at-position (bool) = Ensure that multiple probe sets at a given genomic location have the same CN state
    default value: 'True'

  --kernel-sigma-span (float64) = Probes spanned by one standard deviation
    default value: '50.000000'

  --@help       = print developer help

</pre></div> </div>
<hr class="footer"/><address class="footer"><small>Generated on Monday March 11 2024 12:27:00 for Analysis Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
