<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE clshtml PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
      "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type"
  content="application/xhtml+xml; charset=UTF-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=9" />
  <title>Analysis Power Tools: MANUAL: apt-copynumber-axiom-ref (2.11.8)</title>
  <link href="tabs.css" rel="stylesheet" type="text/css" />
  <script type="text/javascript" src="jquery.js">
  </script>
  <script type="text/javascript" src="dynsections.js">
  </script>
  <link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>

<body>

<div id="top">
<!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->

<div id="navrow1" class="tabs">
<ul class="tablist">
  <li><a href="index.html"><span>Main Page</span></a></li>
  <li><a href="CHANGE.html"><span>Change Log</span></a></li>
  <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
  <li><a href="FAQ.html"><span>FAQs</span></a></li>
  <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
  <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
</ul>
</div>
</div>
<!-- top -->

<div class="header">

<div class="headertitle">

<div class="title">
<h1>MANUAL: apt-copynumber-axiom-ref (2.11.8) </h1>
</div>
</div>
</div>
<!--header-->

<div class="contents">

<div class="textblock">
<h2>Contents </h2>
<ul>
  <li><a href="#intro">Introduction</a></li>
  <li><a href="#quickstart">Quick Start</a></li>
  <li><a href="#options">Options</a></li>
</ul>

<p><a class="anchor" id="intro"></a> </p>

<h2>Introduction </h2>

<p><b>apt-copynumber-axiom-ref</b> supports the creation of both standard and a
region-based reference model file for downstream use in apt-copynumber-hmm and
apt-copynumber-cnvmix executables. Reference generation can be started either
from a set of CEL files or from a set of files from a previously run
apt-genotyping-axiom run. The algorithm assumes that all CEL files have passed
sample QC and are copy number neutral for both the autosomal and sex
chromosomes.</p>

<p><a class="anchor" id="quickstart"></a> </p>

<h2>Quick Start </h2>
<p>An example command with input cel_list.txt is shown below. This file has one column with the header ‘cel_files’ and a list of all input CEL files with full paths specified. </p>
<pre class="fragment">
  apt-copynumber-axiom-ref \
  --analysis-files-path library_directory \
  --arg-file library_directory/arrayname.r1.apt-copynumber-axiom-ref.AxiomCNref.apt2.xml \
  --apt-axiom-cnv-node:annotation-file library_directory/arrayname.na36.r1.annot.db \
  --cel-files cel_list.txt \
  --out-dir output_directory \
  --log-file output_directory/apt-copynumber-axiom-ref.log \
  --new-reference-file output_directory/arrayname.r1.cn_models
</pre>
<p>If an annotation file is specified in the .xml file and exists in --analysis-files-path, then the --apt-axiom-cnv-node:annotation-file option is not required in the script.</p>
<p>For other usage example and tips for creating a reference, see the Axiom Copy Number Data Analysis Guide.</p>
<p><a class="anchor" id="options"></a> </p>

<h2>Options</h2>

<div class="fragment">
<pre class="fragment">'apt-copynumber-axiom-ref'
   'apt-copynumber-axiom-ref.exe'
    APT_2.11.8-rc4 https://github.com/thermofisher/ma-alg-apt2-genotyping f1c1c38(a9a90c6) 2023-03-21


apt-copynumber-axiom-ref supports the creation of both generic and a region-based reference model files.
Reference generation can be started either from a set of cel files or from a set of files from a previously run apt-genotype-axiom analysis.

USAGE:
Generic run from cel files
apt-copynumber-axiom-ref --out-dir [directory] --cel-files [cel files file] --dual-channel-normalization true --arg-file [additional args] \
--analysis-files-path [library dir] --allele-summaries true --apt-axiom-cnv-node:annotation-file [annotation file] \
--special-snps [.specialSNPs file] --new-reference-file [output reference file] --reference-file-template [cn_models_template file]

Generic run using files from a previous apt-genotyping-axiom run. --skip-genotyping = true
apt-copynumber-axiom-ref --out-dir [directory] --summary-file [summary file]  --adjusted-intensities-node:calls-file [calls file] \
--cn-library-file-node:report-file [report file] --apt-axiom-cnv-node:annotation-file [annotation file] --skip-genotyping true \
--special-snps-file [.specialSNPs file] --new-reference-file [output reference file] --reference-file-template [cn_models_template file]

Region-based run has the same set of parameters as the generic run but an additional parameter must be set --fixed-cn-region-calls-file.

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    <Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" />.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --delete-temp-files (bool) = Delete the temporary files after the run completes.
    default value: 'True'

  --analysis-files-path (string8) = Default directory to search for analysis library files

  --new-reference-file (filename,required) = The cn_models file that is to be created

  --reference-file-template (filename,required) = The cn_models_template file based upon which the cn_models file should be created

  --skip-genotyping (bool) = skip the apt-genotyping-axiom step
    default value: 'False'

  --use-wave-correction (bool) = Use wave correction
    default value: 'False'

  --human-X-chromosome (string8) = Defines the X chromosome index
    default value: 'X'

  --human-Y-chromosome (string8) = Defines the Y chromosome index
    default value: 'Y'

  --add-AD-median-signals (bool) = AD median signals will be calculated if true
    default value: 'False'

  --intensity-covariate-flags (filename) = A key-value covariate-flag text file overriding the hardcoded mapping of intensity covariate names to process flags

  --lr-covariate-flags (filename) = A key-value covariate-flag text file overriding the hardcoded mapping of log2 ratio covariate names to process flags

  --apcov-order (string8) = Names of the allele peaks covariates in the order that they are to be applied.

  --apcov-bin-type (string8) = Bin type for each covariate (equally-populated, equally-spaced, discrete).

  --apcov-bin-count (string8) = Bin count for each covariate.

  --apcov-coarse-allele-peak-adjustment (string8) = Apply coarse allele peak adjustment for each covariate (on, off).
    default value: 'on'

  --apcov-coarse-allele-peak-adjustment-step (string8) = 
    default value: '20'

  --apcov-coarse-allele-peak-adjustment-window (string8) = 
    default value: '100'

  --apcov-coarse-allele-peak-adjustment-point-count (string8) = 
    default value: '128'

  --apcov-coarse-allele-peak-adjustment-bandwidth (string8) = 
    default value: '0.25'

  --apcov-coarse-allele-peak-adjustment-cutoff (string8) = 
    default value: '0.05'

  --apcov-coarse-allele-peak-adjustment-clean-threshold (string8) = 0.35

  --apcov-coarse-allele-peak-adjustment-height-threshold (string8) = 
    default value: '0.0'

  --apcov-coarse-allele-peak-adjustment-height-threshold-bound (string8) = 
    default value: '1024.0'

  --apcov-coarse-allele-peak-adjustment-outlier-trim (string8) = 
    default value: '3.0'

  --apcov-covariate-bin-peaks-point-count (string8) = 

  --apcov-covariate-bin-peaks-bandwidth (string8) = 

  --apcov-master-peaks-point-count (string8) = 

  --apcov-master-peaks-bandwidth (string8) = 

  --apcov-kernel-function-selection (string8) = 
    default value: 'gaussian'

  --apcov-disable (string8) = Use this to disable allele-peak covariate adjustment. Removing all apcov-* parameters will have the same effect.
    default value: 'true'

  --mapd-max (float64) = Samples with MAPD greater than this threshold will fail quality control
    default value: '0.400000'

  --waviness-sd-max (float64) = Samples with waviness standard deviation greater than this threshold will fail quality control
    default value: '0.300000'

  --unadjusted-intensities-node:out-dir (string8) = The output directory for result files.

  --temp-dir (string8) = The output directory for temporary files.

  --unadjusted-intensities-node:analysis-files-path (dirname) = Default directory to search for analysis library files

  --force (bool) = Force the files to process regardless of array type.
    default value: 'False'

  --chip-type (vec_string8,required) = Text identifier for chip.
    Aliases: --array-type

  --sketch-target-output (bool) = Write the quantile normalization distribution (or sketch) to a file for reuse with target-sketch option. [default 'false']
    default value: 'False'
    Aliases: --write-sketch

  --sketch-target-output-file (string8) = Optional fully-specified path of sketch.txt file.

  --dual-channel-target-sketch (bool) = Create sketches from separate channels (AT/GC)
    default value: 'True'

  --sketch-target-input-file (string8) = Name of the file containing sketch to use.
    Aliases: --target-sketch

  --sketch-target-scale-value (float64) = Target to scale the sketch to
    default value: '0.000000'

  --dual-channel-normalization (bool) = Use dual channel normalization.
    default value: 'True'

  --probeset-ids (string8) = Tab delimited file with column 'probeset_id' specifying probesets to genotype.

  --library-file-node:special-snps (string8) = File specifying non-autosomal chromosome information.

  --cel-files (string8,required) = The file that identifies the cel files to process.

  --raw-gender-write-intensities-node:normalized-intensities-output-file (filename) = Optional full path of normalized intensities output file.  I.e. Use this filename instead of IntensityScratchPadFileName_*.

  --normalization-write-intensities-node:normalized-intensities-output-file (filename) = Optional full path of normalized intensities output file.  I.e. Use this filename instead of IntensityScratchPadFileName_*.

  --allele-summaries (bool) = Print file of intensities summarized by allele.
    default value: 'False'
    Aliases: --summaries

  --genotyping-node:a5-tables (bool) = Output calls, confidences in hdf5 format.
    default value: 'False'

  --summary-file (filename) = The Axiom summary file

  --adjusted-intensities-node:calls-file (filename) = The Axiom calls file

  --cn-library-file-node:report-file (filename) = The report file generated by apt-genotype-axiom (used as input)

  --cn-library-file-node:annotation-file (filename) = 
    default value: 'Annotation SQL file.'

  --samples-per-block (int32) = Number of samples per block to load from summary HDF5 file. Default is 200.
    default value: '200'

  --apt-axiom-cnv-node:annotation-file (filename,required) = Annotation file

  --apt-axiom-cnv-node:min-number-male-samples (int32) = The minimum recommended number of male samples
    default value: '40'

  --apt-axiom-cnv-node:min-number-female-samples (int32) = The minimum recommended number of female samples
    default value: '40'

  --fixed-cn-region-calls-file (filename) = Input file listing expected copy number states by fixed region and cel file.

  --normal-cn-fixed-regions-file (filename) = Optional file listing normal CN state(s) for CN_Region, so that samples with same copy number state(s) in optional fixed-cn-region-calls-file are used for reference generation. File has column names labeled CN_Region and CN_State_Normal.

  --qc-results-file (string8) = Path to report file of raw MAPD and WavinessSD results.  Filename only assumes out-dir directory.

  --wave-trim (float64) = The trim to apply to the log2 ratios.
    default value: '2.000000'

  --wave-percentile (float64) = The high percentile value to apply.
    default value: '0.750000'

  --wave-demean (bool) = Demean the input to the SVD.
    default value: 'False'

  --wave-count (int32) = The number of waves to calculate.
    default value: '-1'

  --wave-count-offset (int32) = The number of waves to add beyond plate count, when computing wave-count instead of using --wave-count option
    default value: '1'

  --max-wave-count (int32) = The upper limit on number of waves to to save to models file.
    default value: '20'

  --wave-correction-method (string8) = The method to be used to compute waves. Either "pca-like" or "pca". The method "pca-like" uses a subset of probe sets to reduce memory requirements. "pca" uses the standard principal component analysis.
    default value: 'pca'

  --do-write-scores (bool) = Enable scores.txt output
    default value: 'True'

  --@help       = print developer help

</div>
</div>
</div>
<!-- contents -->
<!-- start footer part -->
<hr class="footer" />
<address class="footer">
  <small>Generated on Tuesday March 28 2022 12:20:00 for Analysis Power Tools by&nbsp; <a
  href="http://www.doxygen.org/index.html"><img class="footer"
  src="doxygen.png" alt="doxygen" /> </a> 1.8.5 </small> 
</address>
</body>
</html>
