<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Analysis Power Tools: MANUAL: apt-genotype-axiom (2.12.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt-genotype-axiom (2.12.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
<li><a href="#references">References</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>apt-genotype-axiom</b> is a program for making genotyping calls on the Affymetrix Axiom&trade; family of arrays. <b>apt-genotype-axiom</b> uses the BRLMM-P algorithm for genotype calling. BRLMM-P is a model-based approach which performs 1-dimensional clustering by fitting a Gaussian mixture model. Users wishing to use other legacy genotyping algorithms are recommended to use the <code>apt-probeset-genotype</code> application.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<p>The command below runs the Axiom&trade; GT1 algorithm on Axiom&trade; arrays.</p>
<p> Note: The commands listed below are for standard biallelic analysis, without CNAG.  For additional information on rare het adjustment, see FAQs below.</p>
<pre class="fragment">
apt-genotype-axiom \
  --analysis-files-path /library/file/path \
  --arg-file Axiom_PMRA_96orMore_Step2.r3.apt-genotype-axiom.AxiomGT1.apt2.xml \
  --do-rare-het-adjustment true \
  --out-dir out \
  --cel-files cel_file_list.txt \
  --log-file ./apt-genotype-axiom.log
</pre><p>Alternatively, parameters can be set using command line parameters directly:</p>
<pre class="fragment">
apt-genotype-axiom \
  --analysis-files-path /library/file/path \
  --analysis-name AxiomGT1 \
  --chip-type Axiom_PMRA.r3 \
  --do-rare-het-adjustment true \
  --artifact-reduction-clip 0.4 \
  --artifact-reduction-close 2 \
  --artifact-reduction-open 2 \
  --artifact-reduction-fringe 4 \
  --artifact-reduction-cc 2 \
  --sketch-target-scale-value 1000 \
  --sketch-size 50000 \
  --genotyping-node:brlmmp-CM 1 \
  --genotyping-node:brlmmp-bins 100 \
  --genotyping-node:brlmmp-mix 1 \
  --genotyping-node:brlmmp-bic 2 \
  --genotyping-node:brlmmp-lambda 1.0 \
  --genotyping-node:brlmmp-HARD 3 \
  --genotyping-node:brlmmp-SB 0.75 \
  --genotyping-node:brlmmp-transform MVA \
  --genotyping-node:brlmmp-copyqc 0.00000 \
  --genotyping-node:brlmmp-wobble 0.05 \
  --genotyping-node:brlmmp-MS 0.15 \
  --genotyping-node:brlmmp-copytype -1 \
  --genotyping-node:brlmmp-clustertype 2 \
  --genotyping-node:brlmmp-ocean 0.00001 \
  --genotyping-node:brlmmp-CSepPen 0.1 \
  --genotyping-node:brlmmp-CSepThr 4 \
  --genotyping-node:snp-priors-input-file Axiom_PMRA.r3.generic_prior.txt \
  --probeset-ids Axiom_PMRA.r3.step2.ps \
  --cdf-file Axiom_PMRA.r3.cdf \
  --special-snps Axiom_PMRA.r3.specialSNPs \
  --x-probes-file Axiom_PMRA.r3.chrXprobes \
  --y-probes-file Axiom_PMRA.r3.chrYprobes \
  --sketch-target-input-file Axiom_PMRA.r3.AxiomGT1.sketch \
  --igender-female-threshold 1.05 \
  --igender-male-threshold 1.1 \
  --allele-summaries true \
  --genotyping-node:snp-posteriors-output true \
  --out-dir out \
  --cel-files cel_file_list.txt \
  --log-file ./apt-genotype-axiom.log
</pre>
<p><a class="anchor" id="options"></a></p>
<h2>Options</h2>
<pre class="fragment">
   'apt-genotype-axiom.exe'
   v2.12.0-rc2 https://github.com/thermofisher/ma-alg-apt2-genotyping 483f659(80ecaa3) 2023-09-12

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    &lt;Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" /&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --max-threads (int32) = Maximum thread count (0 = unlimited threads); Default = number of CPUs detected
    default value: '0'
    Aliases: --mt, --max_threads

  --delete-temp-files (bool) = Delete the temporary files after the run completes.
    default value: 'True'

  --out-dir (string8) = The output directory for result files.

  --temp-dir (string8) = The output directory for temporary files.

  --analysis-files-path (dirname) = Default directory to search for analysis library files

  --batch-folder (string8) = Full path to Agatha batch folder. All genotyping data will be streamed into this folder as'all_genotypes_by_snps.CHP.bin' and 'all_genotypes_by_snps.CHP.index.txt'.

  --batch-folder-data-dir (string8) = Name of directory inside batch-folder where the analysis suite data is written.
    default value: 'AxiomAnalysisSuiteData'

  --force (bool) = Force the files to process regardless of array type.
    default value: 'False'

  --chip-type (vec_string8,required) = Text identifier for chip.
    Aliases: --array-type

  --analysis-name (string8) = Name to prepend to output files.

  --report (bool) = Print report.txt file of summary and genotyping statistics.
    default value: 'True'

  --report-file (string8) = Optional full path of report.txt file.

  --summaries-only (bool) = Skip the genotyping step.
    default value: 'False'

  --x-probes-file (filename) = The file that identifies the X chromosome probe ids.

  --y-probes-file (filename) = The file that identifies the Y chromosome probe ids.

  --w-probes-file (filename) = The file that identifies the W chromosome probe ids.

  --z-probes-file (filename) = The file that identifies the Z chromosome probe ids.

  --sketch-target-output (bool) = Write the quantile normalization distribution (or sketch) to a file for reuse with target-sketch option. [default 'false']
    default value: 'False'
    Aliases: --write-sketch

  --sketch-target-output-file (string8) = Optional fully-specified path of sketch.txt file.

  --dual-channel-target-sketch (bool) = Create sketches from separate channels (AT/GC)
    default value: 'True'

  --sketch-target-input-file (string8) = Name of the file containing sketch to use.
    Aliases: --target-sketch

  --sketch-target-scale-value (float64) = Target to scale the sketch to
    default value: '0.000000'

  --sketch-target-use-avg (bool) = When there is a range of identical intensities, use the average of the corresponding target sketch values.
    default value: 'True'

  --dual-channel-normalization (bool) = Use dual channel normalization.
    default value: 'True'

  --probabilities-output (bool) = Output a file with comma-separated probabilities for BB,AB,AA,Ocean for each probeset and sample [default 'false']
    default value: 'False'

  --summary-a5-output (bool) = Print summary in a5 format instead of plain text.
    default value: 'False'

  --do-rare-het-adjustment (bool) = Analyze probe performance for rare biallelic het calls.  Conditionally change to NoCall.
    default value: 'False'

  --rare-het-candidate-file (string8) = File of probeset-ids and CEL name pairs that indicate candidates for rare het analysis

  --process-multi-alleles (bool) = Augment genotyping workflow to accomodate multi-allelic snps (allele count > 2)
    default value: 'False'

  --probeset-ids (string8) = Tab delimited file with column 'probeset_id' specifying probesets to genotype.

  --cdf-file (string8) = The CDF input file.

  --special-snps (string8) = File specifying non-autosomal chromosome information.

  --snp-specific-param-file (string8) = File specifying SNP-specific paramaters overriding standard parameters in genotyping.
    File must have columns: ProbeSetName, ShellBarrier, Ocean, Confidence, CSepPen and CSepThr.  File may also have optional final column: Lambda.
    Aliases: --ssp-file

  --cel-files (string8,required) = The file that identifies the cel files to process.

  --gender-file (string8) = File specifying genders for each CEL file.
    File must have columns cel_files and gender.
    Aliases: --read-genders

  --inbred-file (string8) = File specifying inbred het penalty for each CEL file.
    File must have columns cel_files and inbred_het_penalty.
    Aliases: --read-inbred

  --hints-file (string8) = File specifying seed genotypes instead of DM algorithm
    File must have first column 'probeset_id' and one additional column per CEL file, with file basename as column name.
    Rows must have probeset id for first column and integers corresponding to calls for remaining columns.
    Calls represented by 'No Call' = -1, 'AA' = 0, 'AB' = 1, 'BB' = 2.
    Aliases: --genotypes

  --global-inbred-het-penalty (float64) = Specifying global inbred het penalty value for each CEL file.

    default value: '0.000000'
    Aliases: --global-inbred

  --igender-female-threshold (float64) = The female threshold to use in the raw intensity gender call.
    default value: '0.480000'
    Aliases: --female-thresh

  --igender-male-threshold (float64) = The male threshold to use in the raw intensity gender call.
    default value: '0.700000'
    Aliases: --male-thresh

  --artifact-reduction-clip (float64) = Threshold for intensity log-ratio to be outlier
    default value: '0.400000'
    Aliases: --artifact-reduction-clip-qcam

  --artifact-reduction-clip-pvcam (float64) = Threshold for intensity log-ratio to be outlier scan image is from PVCam
    default value: '0.430000'

  --artifact-reduction-open (int32) = Minimum size of snow (isolated residuals) to remove
    default value: '1'

  --artifact-reduction-close (int32) = Max gap to close between tentative artifacts
    default value: '3'

  --artifact-reduction-fringe (int32) = Add a fringe to catch border features that might be poor.
    default value: '2'

  --artifact-reduction-cc (int32) = How many channels must support an artifact
    default value: '1'

  --artifact-reduction-trustcheck (bool) = Completely untrusted, blemished probesets in samples are no-calls
    default value: 'False'

  --artifact-reduction-output-trustcheck (bool) = Output text file of blemished probesets with cel index
    default value: 'False'
    Aliases: --output-trustcheck

  --raw-gender-write-intensities-node:normalized-intensities-output-file (filename) = Optional full path of normalized intensities output file.  I.e. Use this filename instead of IntensityScratchPadFileName_*.

  --sketch-size (int32,required) = The sketch size to use.

  --normalization-write-intensities-node:normalized-intensities-output-file (filename) = Optional full path of normalized intensities output file.  I.e. Use this filename instead of IntensityScratchPadFileName_*.

  --blob-format-version (float32) = Blob format version for storing genotyping result. 1.0=legacy, 2.0=multiallelic/copynumber.
    default value: '2.000000'

  --sample-multiplier (int32) = Used for generating fake blob with large number of samples. Ex: 10 samples x 500 multiplier = 5000 fake samples.
    default value: '1'

  --use-copynumber-call-codes (bool) = Use copynumber aware call codes, instead of raw brlmmp call codes.
    default value: 'False'

  --allele-summaries (bool) = Print file of intensities summarized by allele.
    default value: 'False'
    Aliases: --summaries

  --allele-summaries-file (string8) = Optional fully-specified path of allele-summaries.txt file.
    Aliases: --summaries-file

  --background-percentile (float64) = The percentile obtained from the signals for a probeset.
    default value: '0.100000'
    Aliases: --bgnd-percentile

  --pedestal (float64) = A given value by which to divide the background percentile.
    default value: '256.000000'
    Aliases: --bgnd-pedestal

  --genotyping-node:snp-priors-input-file (string8,required) = The string that identifies the snp priors to use when genotyping.

  --genotyping-node:snp-posteriors-output (bool) = Write snp posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-models

  --genotyping-node:snp-posteriors-output-file (string8) = Optional full path of snp-posteriors.txt file.

  --genotyping-node:table-output (bool) = Output matching matrices of tab delimited genotype calls and confidences. [default 'true']
    default value: 'True'

  --genotyping-node:a5-tables (bool) = Output calls, confidences in hdf5 format.
    default value: 'False'

  --genotyping-node:calls-file (string8) = Optional full path of calls.txt file.

  --genotyping-node:confidences-file (string8) = Optional full path of confidences.txt file.

  --genotyping-node:probabilities-file (string8) = Optional full path of probabilities template file.  The file is typically partitioned and integers inserted into the filename before the .txt suffix.

  --genotyping-node:probabilities-file-sample-count (int32) = Number of samples per probability file.  These files can be large, especially when running 1000's of samples. [default '0' = don't break up the file]
    default value: '0'
    Aliases: --prob-file-sample-count

  --genotyping-node:copynumber-probeset-calls-file (string8) = Optional full path of probeset copynumber calls file.

  --genotyping-node:copynumber-override-special-snps (bool) = Replace specialSnp copynumber gender designation with copynumber call
    default value: 'False'

  --genotyping-node:brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --genotyping-node:brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --genotyping-node:brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '1'

  --genotyping-node:brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --genotyping-node:brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --genotyping-node:brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --genotyping-node:brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --genotyping-node:brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --genotyping-node:brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --genotyping-node:brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --genotyping-node:brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --genotyping-node:brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --genotyping-node:brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --genotyping-node:brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --genotyping-node:brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --genotyping-node:brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --genotyping-node:brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --genotyping-node:brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '16.000000'

  --genotyping-node:brlmmp-comvar (bool) = boolean to indicate if common variance is used in all clusters.
    default value: 'True'

  --genotyping-node:brlmmp-K (float64) = Scaling Parameter to use in transformations.
    default value: '1.000000'

  --genotyping-node:max-rare-het-count (int32) = Max number of het calls for a probeset over all samples to trigger SCI analysis.
    default value: '3'

  --genotyping-node:gc-channel-sci-threshold (float64) = Max threshold for SCI values from intensities in the GC channel.
    default value: '0.390000'

  --genotyping-node:at-channel-sci-threshold (float64) = Max threshold for SCI values from intensities in the AT channel.
    default value: '0.390000'

  --genotyping-node:major-hom-relative-min-percent (float64) = Minimum percentage of calls that are hom in order to meet major hom criteria.
    default value: '80.000000'

  --genotyping-node:major-hom-absolute-min-thresh (int32) = Minimum number of calls that are hom in order to meet major hom criteria.
    default value: '15'

  --multi-allele-background-node:snp-priors-input-file (string8,required) = The string that identifies the snp priors to use when genotyping.

  --multi-allele-background-node:snp-posteriors-output (bool) = Write snp posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-models

  --multi-allele-background-node:snp-posteriors-output-file (string8) = Optional full path of snp-posteriors.txt file.

  --multi-allele-background-node:table-output (bool) = Output matching matrices of tab delimited genotype calls and confidences. [default 'true']
    default value: 'True'

  --multi-allele-background-node:a5-tables (bool) = Output calls, confidences in hdf5 format.
    default value: 'False'

  --multi-allele-background-node:calls-file (string8) = Optional full path of calls.txt file.

  --multi-allele-background-node:confidences-file (string8) = Optional full path of confidences.txt file.

  --multi-allele-background-node:copynumber-probeset-calls-file (string8) = Optional full path of probeset copynumber calls file.

  --multi-allele-background-node:copynumber-override-special-snps (bool) = Replace specialSnp copynumber gender designation with copynumber call
    default value: 'False'

  --multi-allele-background-node:brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --multi-allele-background-node:brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --multi-allele-background-node:brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '1'

  --multi-allele-background-node:brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --multi-allele-background-node:brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --multi-allele-background-node:brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --multi-allele-background-node:brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --multi-allele-background-node:brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --multi-allele-background-node:brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --multi-allele-background-node:brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --multi-allele-background-node:brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --multi-allele-background-node:brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --multi-allele-background-node:brlmmp-transform (string8) = brlmmp transform. (mva, rvt, ssf, ces, assf, ccs)
    default value: 'mva'

  --multi-allele-background-node:brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --multi-allele-background-node:brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '16.000000'

  --multi-allele-background-node:brlmmp-comvar (bool) = boolean to indicate if common variance is used in all clusters.
    default value: 'True'

  --multi-allele-background-node:brlmmp-K (float64) = Scaling Parameter to use in transformations.
    default value: '1.000000'

  --multi-allele-background-node:max-rare-het-count (int32) = Max number of het calls for a probeset over all samples to trigger SCI analysis.
    default value: '3'

  --multi-allele-background-node:gc-channel-sci-threshold (float64) = Max threshold for SCI values from intensities in the GC channel.
    default value: '0.390000'

  --multi-allele-background-node:at-channel-sci-threshold (float64) = Max threshold for SCI values from intensities in the AT channel.
    default value: '0.390000'

  --multi-allele-background-node:major-hom-relative-min-percent (float64) = Minimum percentage of calls that are hom in order to meet major hom criteria.
    default value: '80.000000'

  --multi-allele-background-node:major-hom-absolute-min-thresh (int32) = Minimum number of calls that are hom in order to meet major hom criteria.
    default value: '15'

  --multi-allele-pairwise-node:snp-priors-input-file (string8,required) = The string that identifies the snp priors to use when genotyping.

  --multi-allele-pairwise-node:snp-posteriors-output (bool) = Write snp posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-models

  --multi-allele-pairwise-node:snp-posteriors-output-file (string8) = Optional full path of snp-posteriors.txt file.

  --multi-allele-pairwise-node:table-output (bool) = Output matching matrices of tab delimited genotype calls and confidences. [default 'true']
    default value: 'True'

  --multi-allele-pairwise-node:a5-tables (bool) = Output calls, confidences in hdf5 format.
    default value: 'False'

  --multi-allele-pairwise-node:calls-file (string8) = Optional full path of calls.txt file.

  --multi-allele-pairwise-node:confidences-file (string8) = Optional full path of confidences.txt file.

  --multi-allele-pairwise-node:copynumber-probeset-calls-file (string8) = Optional full path of probeset copynumber calls file.

  --multi-allele-pairwise-node:copynumber-override-special-snps (bool) = Replace specialSnp copynumber gender designation with copynumber call
    default value: 'False'

  --multi-allele-pairwise-node:brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --multi-allele-pairwise-node:brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --multi-allele-pairwise-node:brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '1'

  --multi-allele-pairwise-node:brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --multi-allele-pairwise-node:brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --multi-allele-pairwise-node:brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --multi-allele-pairwise-node:brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --multi-allele-pairwise-node:brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --multi-allele-pairwise-node:brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --multi-allele-pairwise-node:brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --multi-allele-pairwise-node:brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --multi-allele-pairwise-node:brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --multi-allele-pairwise-node:brlmmp-transform (string8) = brlmmp transform. (mva, rvt, ssf, ces, assf, ccs)
    default value: 'mva'

  --multi-allele-pairwise-node:brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --multi-allele-pairwise-node:brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '16.000000'

  --multi-allele-pairwise-node:brlmmp-comvar (bool) = boolean to indicate if common variance is used in all clusters.
    default value: 'True'

  --multi-allele-pairwise-node:brlmmp-K (float64) = Scaling Parameter to use in transformations.
    default value: '1.000000'

  --multi-allele-pairwise-node:max-rare-het-count (int32) = Max number of het calls for a probeset over all samples to trigger SCI analysis.
    default value: '3'

  --multi-allele-pairwise-node:gc-channel-sci-threshold (float64) = Max threshold for SCI values from intensities in the GC channel.
    default value: '0.390000'

  --multi-allele-pairwise-node:at-channel-sci-threshold (float64) = Max threshold for SCI values from intensities in the AT channel.
    default value: '0.390000'

  --multi-allele-pairwise-node:major-hom-relative-min-percent (float64) = Minimum percentage of calls that are hom in order to meet major hom criteria.
    default value: '80.000000'

  --multi-allele-pairwise-node:major-hom-absolute-min-thresh (int32) = Minimum number of calls that are hom in order to meet major hom criteria.
    default value: '15'

  --multi-freq-flag (bool) = Adjustment factor for the frequency of the cluster.
    default value: 'False'

  --multi-inflate-PRA (float64) = Inflate amount of the posteriors covariance matrices
    default value: '0.000000'

  --multi-ocean (float64) = Analogous to the brlmmp ocean parameter but used in nulti-allele genotyping.
    default value: '0.000010'

  --multi-lambda-P (float64) = Coefficient for pooling common variance.
    default value: '0.000000'

  --multi-wobble (float64) = 
    default value: '0.050000'

  --multi-copy-distance-0to1 (float64) = Used to adjust empty clusters positions based on the nonempty cluster data
    default value: '1.500000'

  --multi-copy-distance-1to2 (float64) = Used to adjust empty clusters positions based on the nonempty cluster data
    default value: '0.200000'

  --multi-shell-barrier-0to1 (float64) = Push copy number 0 clusters away from copy number 1 clusters.
    default value: '0.050000'

  --multi-shell-barrier-1to2 (float64) = Push the copy number 2 cluster away from copy number 1 clusters.
    default value: '0.050000'

  --multi-confidence-threshold (float64) = Assign NoCall if the genotyping confidence is larger than this number.
    default value: '0.150000'

  --multi-priors-input-file (string8,required) = The string that identifies the multi-allelic priors to use when genotyping.

  --multi-posteriors-output (bool) = Write multi-allelic posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-multi-models

  --multi-posteriors-output-file (string8) = Full path of multi-allelic posteriors file.

  --@help       = print developer help

</pre><p><a class="anchor" id="faq"></a> </p>
<h2>Frequently Asked Questions </h2>
<p><B>Question</B>: Is there something I can do to improve the accuracy of rare heterozygous calls?</p>
<B>Answer:</B> If you are genotyping a human array with batches of at least 90 samples, you may want to set the parameter 
--do-rare-het-adjustment TRUE. 
When this parameter is TRUE, then heterozygous calls for some probesets where the count of initial heterozygous calls is between 1 and --genotyping-node:max-rare-het-count will have additional tests applied.  
If the heterozygous call does not pass the additional tests, then it will be changed to NoCall.

<p><B>Question:</B> What are the additional output files if do-rare-het-adjustment is turned on?</p>
<B>Answer:</B> AxiomGT1.rare_het.report.txt, which contains all (probeset, cel file) pairs in the batch for which the probeset was identified as having rare hets. 
It includes whether the call was changed to NoCall. The AxiomGT1.rare_het_maj_hom.report.txt contains homozygous cluster statistics for all probesets identified as 
having rare heterozygous calls.


<p><a class="anchor" id="references"></a> </p>
<h2>References </h2>
<ul>
<li>BRLMM-P - <a href="http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf">http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf</a></li>
<li>PLIER - <a href="http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf">http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf</a> </li>
</ul>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Monday March 11 2024 12:27:00 for Analysis Power Tools by&nbsp;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
