<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
      "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
<head>
  <meta http-equiv="Content-Type"
  content="application/xhtml+xml; charset=UTF-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=9" />
  <title>Analysis Power Tools: MANUAL: apt-copynumber-axiom-cnvmix
  (2.12.0)</title>
  <link href="tabs.css" rel="stylesheet" type="text/css" />
  <script type="text/javascript" src="jquery.js">
 </script>
  <script type="text/javascript" src="dynsections.js">
 </script>
  <link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>

<body>

<div id="top">
<!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->

<div id="navrow1" class="tabs">
<ul class="tablist">
  <li><a href="index.html"><span>Main Page</span></a></li>
  <li><a href="CHANGE.html"><span>Change Log</span></a></li>
  <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
  <li><a href="FAQ.html"><span>FAQs</span></a></li>
  <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
  <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
</ul>
</div>
</div>
<!-- top -->

<div class="header">

<div class="headertitle">

<div class="title">
<h1>MANUAL: apt-copynumber-axiom-cnvmix (2.12.0) </h1>
</div>
</div>
</div>
<!--header-->

<div class="contents">

<div class="textblock">
<h2>Contents </h2>
<ul>
  <li><a href="#intro">Introduction.</a></li>
  <li><a href="#quickstart">Quick Start</a></li>
  <li><a href="#options">Options</a></li>
</ul>

<p><a class="anchor" id="intro"></a> </p>

<h2>Introduction </h2>

<p><strong>apt-copynumber-cnvmix.exe</strong> is a program that reports copy
number in predefined regions with dynamic log2ratio calling thresholds. The
program also can report on regions lacking heterozygosity (LOH).  Both
features require a cn_models library file with necessary information.”</p>

<p><a class="anchor" id="quickstart"></a> </p>

<h2>Quick Start </h2>

<h2>Command-line Parameters</h2>

<p>apt-copynumber-axiom-cnvmix </p>

<p>--out-dir .\out_dir </p>

<p>--mapd-max 0.35 </p>

<p>--waviness-sd-max 0.1 </p>

<p>--chip-type .\Axiom </p>

<p>--use-text-format-summary-file False </p>

<p>--summary-file .\AxiomGT1.summary.a5 </p>

<p>--report-file .\AxiomGT1.report.txt </p>

<p>--arg-file .AxiomCNVmix.apt2.xml </p>

<p>--analysis-files-path \libraryfilepath</p>

<p>--cn-priors-file .cn_priors </p>

<p>--reference-file .cn_models </p>

<p>--log-file .\logfile.out</p>

<p></p>

<p></p>

<p><a class="anchor" id="options"></a> </p>

<h3>Options</h3>

<div class="fragment">
<pre class="fragment">  'apt-copynumber-axiom-cnvmix.exe'
   v2.12.0-rc1 https://github.com/thermofisher/ma-alg-apt2-genotyping 9342fa1(fff56d4) 2023-08-10
 

   
USER HELP"
--version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    <Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" />.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --out-dir (string8) = Output directory for result files

  --temp-dir (string8) = The output directory for temporary files

  --analysis-files-path (string8) = Default directory to search for analysis library files

  --summary-file (filename) = Summary file generated by apt-genotype-axiom (used as input)

  --reference-file (filename) = Reference file

  --analysis-name (string8) = Name to prepend to output files
    default value: 'AxiomCNVMix'

  --chip-type (vec_string8) = Text identifier for chip.
    Aliases: --array-type

  --mapdc-max (float64) = Samples with MAPDc greater than this threshold will fail quality control. MAPDc may be computed if control samples are used for plate correction
    default value: '0.230000'

  --waviness-sdc-max (float64) = Samples with WavinessSDc greater than this threshold will fail quality control. WavinessSDc may be computed if control samples are used for plate correction
    default value: '0.100000'

  --mapd-max (float64) = Samples with MAPD greater than this threshold will fail quality control if MAPDc is not reported
    default value: '0.400000'

  --waviness-sd-max (float64) = Samples with WavinessSD greater than this threshold will fail quality control if WavinessSDc is not reported
    default value: '0.300000'

  --penn-cnv-output-plate-corrected (bool) = Output plate-corrected log2 ratios and BAFs in Penn CNV format files
    default value: 'False'

  --use-wave-correction (bool) = Use wave correction
    default value: 'False'

  --use-text-format-summary-file (bool) = Use Axiom summary file in text format instead of a5
    default value: 'False'

  --cn-priors-file (filename) = CN priors file

  --compute-plate-corrected-CN-QC (bool) = If true and control samples are used for plate correction, MAPDc and WavinessSDc are computed and the CN QC status of samples will be based on these metrics
    default value: 'False'

  --cn-regions-file (filename) = File that lists the regions for the CN calculation

  --report-file (filename) = Report file generated by apt-genotype-axiom (used as input)

  --controls-file (filename) = The file containing list of control sample file names used for per-plate signal correction

  --samples-per-block (int32) = Number of samples per block to load from summary HDF5 file. Default is 200.
    default value: '200'

  --loh-batch-folder (filename) = The batch folder generated by apt-genotype-axiom (used as input). Set either the loh-batch-folder or loh-calls-file and loh-confidences-file to get LOH segments.

  --loh-calls-file (filename) = The calls file generated by apt-genotype-axiom (used as input)

  --loh-confidences-file (filename) = The confidences file generated by apt-genotype-axiom (used as input)

  --loh-samples-per-block (int32) = Number of samples per block to process from LOH batch folder or calls file. Only supports up to 200.
    default value: '200'

  --loh-error-rate (float64) = Error rate to use in LOH processing
    default value: '0.050000'

  --loh-beta (float64) = Beta to use in LOH processing
    default value: '0.001000'

  --loh-alpha (float64) = Alpha to use in LOH processing
    default value: '0.010000'

  --loh-separation (int32) = Separation to use in LOH processing
    default value: '1000000'

  --loh-min-marker-count (int32) = Minimum marker count to use in LOH processing
    default value: '10'

  --loh-no-call-threshold (float64) = No call threshold to use in LOH processing
    default value: '0.050000'

  --loh-min-genomic-span (int32) = Minimum genomic span to use in LOH processing
    default value: '1000000'

  --wave-bandwidth (int32) = Wave correction bandwidth.
    default value: '101'

  --wave-bin-count (int32) = Wave correction bin count.
    default value: '25'

  --wave-count (int32) = Wave correction wave count.
    default value: '-1'

  --wave-smooth (bool) = Wave correction wave smooth.
    default value: 'False'

  --ap-baf-step (int32) = Step to use in calculating allele peaks
    default value: '25'
    Aliases: --baf-step

  --ap-baf-window (int32) = Window to use in calculating allele peaks
    default value: '150'
    Aliases: --baf-window

  --ap-baf-point-count (int32) = Point count to use in calculating allele peaks
    default value: '129'
    Aliases: --baf-point-count

  --ap-baf-bandwidth (float64) = Bandwidth to use in calculating allele peaks
    default value: '0.300000'
    Aliases: --baf-bandwidth

  --ap-baf-cutoff (float64) = Cutoff to use in calculating allele peaks
    default value: '0.000000'
    Aliases: --baf-cutoff

  --ap-baf-threshold (float64) = Clean threshold to use in calculating allele peaks
    default value: '0.300000'
    Aliases: --baf-threshold

  --ap-baf-height-threshold (float64) = Height threshold factor to use in calculating allele peaks
    default value: '3.000000'
    Aliases: --baf-height-threshold

  --ap-baf-height-threshold-bound (float64) = Bounds between which the height threshold is applied
    default value: '1.000000'
    Aliases: --baf-height-threshold-bound

  --ap-baf-symmetry (bool) = Whether to symmetrize the data when calculating allele peaks
    default value: 'True'
    Aliases: --baf-symmetry

  --shrink-factor-3 (float64) = shrink-factor-3
    default value: '0.500000'

  --shrink-factor-4 (float64) = shrink-factor-4
    default value: '0.600000'

  --fld-weights-value (float64) = Value for the fld weights array
    default value: '100.000000'

  --transform-data (bool) = Whether to transform the data
    default value: 'True'

  --penn-cnv-output (bool) = Output log2 ratios and BAFs in Penn CNV format files
    default value: 'False'

  --cnv-a5-output (bool) = Output log2 ratios and BAFs in HDF5 format files
    default value: 'True'

  --cnvmix-density-point-count (int32) = Number of points used to define kernel density function
    default value: '512'

  --cn-posteriors-file (filename) = CN posteriors file

  --model-prob-min-thr-candidate (float64) = minimum threshold probability of any cluster or plate effect for candidate model
    default value: '0.000100'

  --model-prob-med-thr-candidate (float64) = minimum threshold for median probability of cluster and plate effect for candidate model
    default value: '0.000100'

  --plate-effect-lower-limit (float64) = 
    default value: '-0.300000'

  --plate-effect-upper-limit (float64) = 
    default value: '0.300000'

  --@help       = print developer help
</pre></div>
</div>
</div>
<!-- contents -->
<!-- start footer part -->
<hr class="footer" />
<address class="footer">
  <small>Generated on Monday March 11 2024 12:27:00 for Analysis Power Tools by  <a
  href="http://www.doxygen.org/index.html"><img class="footer"
  src="doxygen.png" alt="doxygen" /> </a> 1.8.5 </small> 
</address>
</body>
</html>
