<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Analysis Power Tools: MANUAL: apt2-pgx-translation (2.12.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt2-pgx-translation (2.12.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p>apt2-pgx-translation engine converts genotype calls for markers of pharmacogenetic interest into Ref/Var call nomenclature, reports haplotype pair names for some genes, and translates the haplotype pairs to predicted enzyme phenotypes. This engine replaces apt-dmet-translation.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<p>The command below is formatted for Linux environment and shows typical usage for a PharmacoScan_96F array with the r9 library package. BATCHDIR is the path to the batch folder created by Axiom Analysis Suite using either the Genotyping or Best Practices workflows. The batch folder can also be created by apt-package-util. AXIOM_LIB_PATH is the path to the PharmacoScan_96F.r9 library folder.
</p>
<div class="fragment"><pre class="fragment">
BASENAME=myresults
apt2-pgx-translation \
--batch-folder $BATCHDIR \
--cn-region-calls-file $BATCHDIR/CNData/AxiomCNVMix.cnregioncalls.txt \
--marker-list-file $BATCHDIR/SNPolisher/Recommended.ps \
--analysis-files-path $AXIOM_LIB_PATH \
--translate-file PharmacoScan_96F.r9.na36.translation \
--metabolizer-file PharmacoScan_96F.r9.metabolizer \
--annotation-file PharmacoScan_96F.r9.dc_annot.csv \
--use-first-dup-allele-def true \
--out-dir $BATCHDIR/Translation \
--base-name-prefix $BASENAME \
--log-file $BATCHDIR/Translation/$BASENAME.log
</pre></div>
<p>To perform allele translation on a subset of samples in the batch, supply the optional --sample-filter-file argument, pointing to the file containing a list of cel filenames (no path).  The first row must be ‘cel_files’.  An example file is shown below:</p>
<div class="fragment"><pre class="fragment">
cel_files
Control1_G12.CEL
Control2_H12.CEL
</pre></div>
<p><a class="anchor" id="options"></a></p>
<h2>Options</h2>
<pre class="fragment">
   'apt2-pgx-translation.exe'
   v2.12.0-rc3 https://github.com/thermofisher/ma-alg-apt2-pgx-translation 2ac7ae0(4d23123) 2023-12-01

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    &lt;Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE"/&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --max-threads (int32) = Maximum thread count (0 = unlimited threads); Default = number of CPUs detected
    default value: '0'
    Aliases: --mt, --max_threads

  --audit (string8) = => Audit various components of translation, don't translate.
Available audit types are:
 * annotation - audit annotation file probe sets against the translation file.

  --use-first-dup-allele-def (bool) = => If marker list filtering removes markers needed to differntiate among multiple possible haplotypes then report only the first named haplotype in the Translation file.
    default value: 'False'

  --annotation-file (string8) = => file.csv (probe set annotation file, CSV)
    Aliases: -a

  --batch-folder (string8) = => Batch folder with AxiomAnalysisSuiteData subfolder that contains genotyping data blob with call codes.
    Aliases: -B

  --marker-list-file (string8) = => marker-list-file.*.txt file is a simple list of probe sets, one per line, to be used for translation.
    Aliases: -m

  --sample-filter-file (string8) = Text file specifying samples or cel files to process, one per line with the first line being 'cel_files'.
    Aliases: -F

  --cn-region-calls-file (string8) = => file.tsv (CN region calls file, TSV)
    Aliases: -R

  --sample-file (string8) = => file.txt (sample info file, TSV)
    Aliases: -s

  --translate-file (string8) = => DMET3_TTTable_vYYYYMMDD_*.txt (DMET2_TTTable_vYYYYMMDD_*.txt for DMET2 testing)
    Aliases: -t

  --metabolizer-file (string8) = The metabolizer bin file for translating allele calls to metabolizer genotypes

  --base-name-prefix (string8) = => Base name prefix of all output reports.
    default value: 'TranslationExperimentReport'
    Aliases: -b

  --regression (int32) =  => Regression log level.
    default value: '0'
    Aliases: -r

  --marker-report (bool) =  => Output DMET2 Marker Report.
    default value: 'False'

  --summary-report-sort (bool) = => Sort the summary report rather than use the default translation table order.
    default value: 'True'

  --temp-dir (string8) = Directory for temporary files when working off disk. Using network mounted drives is not advised. When not set, the output folder will be used. The defaut is typically the output directory or the current working directory.
    default value: '.'

  --out-dir (string8) = Directory for output files. Defaults to current working directory.
    default value: '.'
    Aliases: -o

  --samples-per-block (int32) = Number of samples per block for sample-based scanning.
    default value: '500'

  --analysis-files-path (string8) = Analysis file path.

  --delete-temp-files (bool) = Delete the temporary files after the run completes.
    default value: 'True'

  --@help       = print developer help

</pre>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Monday March 11 2024 12:27:00 for Analysis Power Tools by&nbsp;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
