<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Analysis Power Tools: MANUAL: ps-classification (2.12.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: ps-classification (2.12.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#details">Details</a></li>
<li><a href="#options">Options</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>ps-classification</b> reads in the SNP QC metrics table generated by <a href="ps-metrics.html">ps-metrics</a> and performs SNP
classification based on customizable criteria, including call rate, FLD, HetSO, and HomRO. It classifies
SNPs/probesets into seven major categories: "PolyHighResolution","NoMinorHom", "MonoHighResolution",
"OffTargetVariant", "CallRateBelowThreshold", "Other", and "UnexpectedGenoFreq". The category "Hemizygous" appears only when hemizygous SNPs are in the metrics file but the special SNPs data is not present in the metrics file. `ps-classification` also identifies the best probeset per SNP.</a></p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<p>This command runs ps-classification on a metrics file named metrics.txt  in the directory ./input for a human dataset and writes the classification output to the directory ./classification: </p>
<pre class="fragment">ps-classification --species-type human --metrics-file ./input/metrics.txt --output-dir ./classification
</pre><p><a class="anchor" id="details"></a> </p>
<h2>Details </h2>
<p>The input files and options for ps-classification are:</p>
<ul>
<li>metrics-file - The name of the file containing output generated by ps-metrics.</li>
<li>multi-metrics-file - The name of the multiallelic metrics file generated by ps-metrics.</li>
<li>ps2snp-file(Optional) - The name of a file containing the mapping of probeset IDs to SNP IDs. The column names should be "probeset_id" and "snpid".  The user must supply this file when classifying de novo SNPs or any SNPs that have more than one probeset.</li>
<li>output-dir - Output directory for results files: defaults to current directory.</li>
<li>pid-file- The list of probesets that `ps-classification` should run on.</li>
</ul>
<p>The classification thresholds accepted by ps-classification are:</p>
<ul>
<li>species-type - The type of organism genotyped. The current options are "human", non-human "diploid", and "polyploid".</li>
<li>output-recommended - Logical value indicating whether to output a list of recommended SNPs to recommended.ps.</li>
<li>recommended - List of categories whose SNPs will be output as recommended. If output.recommended is TRUE, the default value for Human is PolyHighResolution, MonoHighResolution, NoMinorHom (legacy data includes Hemizygous); for Diploid the default is PolyHighResolution, MonoHighResolution, NoMinorHom; and for Polyploid the default is PolyHighResolution.</li>
</ul>
<p>If a biallelic metrics file is provided, the following classification thresholds are used:</p>
<ul>
<li>cr-cutoff - Threshold for call rate</li>
<li>fld-cutoff - Threshold for FLD</li>
<li>het-so-cutoff -  Threshold for HetSO</li>
<li>het-so-otv-cutoff - Threshold for OTV detection</li>
<li>hom-ro-1-cutoff - Threshold for HomRO when a SNP has one genotypes</li>
<li>hom-ro-2-cutoff - Threshold for HomRO when a SNP has two genotypes</li>
<li>hom-ro-3-cutoff - Threshold for HomRO when a SNP has three genotypes</li>
<li>hom-ro - Flag indicating whether the metric HomRO is used in classification</li>
<li>hom-het - Flag indicating whether the metric HomHet is used in classification. The HomHet metric identifies two-cluster SNPs/probesets with one homozygote cluster and one heterozygote cluster. This checks if the minor homozygote cluster is missing, which is unreasonable for highly inbred species (e.g. wheat).  This metric should be turned on when classifying SNPs/probesets in highly inbred species.</li>
<li>num-minor-allele-cutoff - Threshold for the number of minor alleles.</li>
<li>priority-order - When performing probeset selection, the best probeset is selected according to the priority order of probeset conversion types. Valid values are sequences of the strings "PolyHighResolution", "NoMinorHom", "OTV", "MonoHighResolution","CallRateBelowThreshold", and "UnexpectedGenoFreq", separated by commas, with each string occuring exactly once: default `'PolyHighResolution,NoMinorHom,OTV,MonoHighResolution,CallRateBelowThreshold'`</li>
<li>genotype-p-value-cutoff - Minimum p-value for a probeset to be recommended instead of UnexpectedGenotypeFrequency</li>
</ul>
<p>The flag <code>hom-ro</code> should be turned off for organisms where the homozygote clusters may center at zero in the contrast space, particularly polyploid species.</p>
<p>If the biallelic metrics file contains copy-number aware metrics, the following classification thresholds are used with SNPs with copy-number aware calls:</p>
<ul>
<li>hom-ro-hap-1-cutoff - Threshold for HomRO when a SNP has one haploid genotype</li>
<li>hom-ro-hap-2-cutoff - Threshold for HomRO when a SNP has one haploid genotype</li>
<li>hom-hap-X-cutoff - Maximum distance diploid clusters can be outside of haploid clusters in the X dimension</li>
<li>hom-hap-Y-lower-cutoff - Maximum distance diploid clusters can be below haploid clusters in the Y dimension</li>
<li>hom-hap-Y-upper-cutoff - Maximum distance haploid clusters can be above diploid clusters in the Y dimension</li>
<li>CN0-hap-X-cutoff - Maximum distance CN0 clusters can be outside of haploid clusters in the X dimension</li>
<li>CN0-hap-Y-cutoff - Maximum distance CN0 clusters can be above haploid clusters in the Y dimension</li>
<li>CN0-dip-X-cutoff - Maximum distance CN0 clusters can be outside of diploid clusters in the X dimension</li>
<li>CN0-dip-Y-cutoff - Maximum distance CN0 clusters can be above diploid clusters in the Y dimension</li>
</ul>
<p>If the supplemental filter is turned on, the following classification thresholds are used with SNPs with supplemental metrics:</p>
<ul>
<li>use-supplemental - Flag indicating if the supplemental classification should be performed</li>
<li>homfld-cut - Minimum HomFLD value for a probeset to be PHR instead of HomHomResolution</li>
<li>n-minor-hom-cut - Minimum number of samples in the minor homozygous cluster to be recommended instead of nMinorHom</li>
<li>hetv-maf-cut - Cut-off for het rate versus MAF for UnexpectedHeterozygosity</li>
<li>variance-class - Category types that the supplemental workflow is run on. Takes a single category or a comma separated list.</li>
<li>bb-varx-cut - Minimum X variance value for the BB cluster to be recommended instead of BBvarianceX</li>
<li>bb-vary-cut - Minimum Y variance value for the BB cluster to be recommended instead of BBvarianceY</li>
<li>ab-varx-cut - Minimum X variance value for the AB cluster to be recommended instead of ABvarianceX</li>
<li>ab-vary-cut - Minimum Y variance value for the AB cluster to be recommended instead of ABvarianceY</li>
<li>aa-varx-cut - Minimum X variance value for the AA cluster to be recommended instead of AAvarianceX</li>
<li>aa-vary-cut - Minimum Y variance value for the AA cluster to be recommended instead of AAvarianceY</li>
<li>bb-varx-z-cut - Minimum z-score X variance value for the BB cluster to be recommended instead of BBvarianceX</li>
<li>bb-vary-z-cut - Minimum z-score Y variance value for the BB cluster to be recommended instead of BBvarianceY</li>
<li>ab-varx-z-cut - Minimum z-score X variance value for the AB cluster to be recommended instead of ABvarianceX</li>
<li>ab-vary-z-cut - Minimum z-score Y variance value for the AB cluster to be recommended instead of ABvarianceY</li>
<li>aa-varx-z-cut - Minimum z-score X variance value for the AA cluster to be recommended instead of AAvarianceX</li>
<li>aa-vary-z-cut - Minimum z-score Y variance value for the AA cluster to be recommended instead of AAvarianceY</li>
<li>bb-varx-z-cut-enabled - Flag indicating if the Z thresholds should be on the BB X variances used instead of the variance thresholds with the supplemental workflow</li>
<li>bb-vary-z-cut-enabled - Flag indicating if the Z thresholds should be on the BB Y variances used instead of the variance thresholds with the supplemental workflow</li>
<li>ab-varx-z-cut-enabled - Flag indicating if the Z thresholds should be on the AB X variances used instead of the variance thresholds with the supplemental workflow</li>
<li>ab-vary-z-cut-enabled - Flag indicating if the Z thresholds should be on the AB Y variances used instead of the variance thresholds with the supplemental workflow</li>
<li>aa-varx-z-cut-enabled - Flag indicating if the Z thresholds should be on the AA X variances used instead of the variance thresholds with the supplemental workflow</li>
<li>aa-vary-z-cut-enabled - Flag indicating if the Z thresholds should be on the AA Y variances used instead of the variance thresholds with the supplemental workflow</li>
</ul>
<p>If the SSPs file is turned on, ps-classification performs the supplemental workflow and the output the SSPs metrics along with the supplemental metrics:</p>
<ul>
<li>use-ssp - Flag indicating if the SSP workflow should be run</li>
</ul>
<p>If the biallelic metrics file contains metrics for special SNPs, the following classifiation thresholds are used with non-PAR X, Y, and MT SNPs:</p>
<ul>
<li>hom-ro-hap-1-MTChr-cutoff - Minimum HomRO value for MT probesets with 1 haploid cluster</li>
<li>hom-ro-hap-2-MTChr-cutoff - Minimum HomRO value for MT probesets with 2 haploid clusters</li>
<li>homfld-YChr-cut - Minimum HomFLD_hap value for a Y probeset to be PHR instead of Other (male samples only)</li>
<li>min-YChr-samples-cut - Minimum number of male and female samples needed to classify Y probesets using the Euclidean distances thresholds</li>
<li>homfld-WChr-cut - Minimum HomFLD_hap value for a W probeset to be PHR instead of Other (female samples only)</li>
<li>min-WChr-samples-cut - Minimum number of male and female samples needed to classify W probesets using the Euclidean distances thresholds</li>
<li>hom-ro-hap-1-XChr-cutoff - Minimum HomRO value for non-PAR X probesets with 1 haploid cluster (male samples only)</li>
<li>hom-ro-hap-2-XChr-cutoff - Minimum HomRO value for non-PAR X probesets with 2 haploid clusters (male samples only)</li>
<li>homfld-XChr-cut - Minimum HomFLD_hap value for a non-PAR X probeset to be PHR instead of Other (male samples only)</li>
<li>fld-XChr-cut - Minimum FLD value for non-PAR X probesets (female samples only)</li>
<li>het-so-XChr-cutoff - Minimum HetSO value for non-PAR X probesets (female samples only)</li>
<li>aaf-XChr-cut - Maximum difference in A-allele frequency between female and male samples for non-PAR X probesets</li>
<li>hom-ro-hap-1-ZChr-cutoff - Minimum HomRO value for Z probesets with 1 haploid cluster (female samples only)</li>
<li>hom-ro-hap-2-ZChr-cutoff - Minimum HomRO value for Z probesets with 2 haploid clusters (female samples only)</li>
<li>homfld-ZChr-cut - Minimum HomFLD_hap value for a Z probeset to be PHR instead of Other (female samples only)</li>
<li>fld-ZChr-cut - Minimum FLD value for Z probesets (male samples only)</li>
<li>het-so-ZChr-cutoff - Minimum HetSO value for Z probesets (male samples only)</li>
<li>aaf-ZChr-cut - Maximum difference in A-allele frequency between female and male samples for Z probesets</li>
</ul>
<p>If a multiallelic metrics file is provided, the following classification thresholds are used:</p>
<ul>
<li>hom-mma-cutoff - Minimum Hom_MMA value needed to be recommended</li>
<li>fld-ma-cutoff - Minimum FLD_MA value needed to be recommended</li>
<li>fld-ma-2-cutoff - Minimum FLD_MA for probesets with 2 alleles</li>
<li>min-fld-ma-cutoff - Minimum FLD_MA value needed to be recommended</li>
<li>min-fld-ma-2-cutoff - Minimum FLD_MA for probesets with 2 alleles</li>
<li>het-so-ma-2-cutoff - Minimum HetSO_MA value needed for probesets wtih 2 alleles to be recommended</li>
<li>hom-ro-ma-cutoff - Minimum HomRO_MA value needed to be recommmended</li>
<li>hom-ro-ma-1-cutoff - Minimum HomRO_MA for probesets with 1 allele</li>
<li>hom-ro-ma-2-cutoff - Minimum HomRO_MA for probesets with 2 alleles</li>
<li>best-cr-cutoff - Minimum CR value needed to be recommended</li>
<li>genotype-p-value-cutoff - Minimum p-value for a probeset to be recommended instead of UnexpectedGenotypeFrequency</li>
</ul>
<p>If a <code>ps2snp-file</code> is provided, then the best performance probeset will be selected to represent SNPs with multiple probesets. There are seven output files, one for each category, containing lists of probeset IDs belonging to each category. Each file contains a list of probeset IDs belonging to the category. The performance output file summarizes the categorization results across all SNPs.</p>
<p>If a <code>psct-file</code> is provided, then probeset-specific classification thresholds can be set.</p>
<p><a class="anchor" id="options"></a> </p>
<h2>Options</h2>
<pre class="fragment">

   'ps-classification.exe'
   v2.12.0-rc3 https://github.com/thermofisher/ma-alg-apt2-snp-qc b13aa60(4d23123) 2024-02-01


USAGE:
   ps-classification --species-type [human|polyploid|diploid] --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
   ps-classification --species-type [human|polyploid|diploid] --metrics-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
   ps-classification --species-type [human|polyploid|diploid] --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]

See the Axiom Data Analysis Guide for more information

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    <Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" />.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --metrics-file (filename) = Full path name for metrics file generated by ps-metrics.

  --species-type (string8,required) = Type of organism to be genotyped. Valid options are 'Human','Diploid' or 'Polyploid'.

  --report-file-ext (string8) = File extension for the ps report files
    default value: 'ps'

  --ps2snp-file (filename) = Full path name of the file mapping probeset IDs to SNP IDs (Recommended).

  --output-dir (dirname) = Path to output directory.

  --output-recommended (bool) = Logical value indicating whether to output a list of recommended SNPs to recommended.ps. Default is TRUE. See the Axiom Data Analysis Guide for more information about SNP conversion.
    default value: 'True'

  --recommended (vec_string8) = List of categories whose SNPs will be output as recommended. If output-recommended is TRUE, the default value for Human is PolyHighResolution, MonoHighResolution, NoMinorHom (legacy data includes Hemizygous); for Diploid the default is PolyHighResolution, MonoHighResolution, NoMinorHom; and for Polyploid the default is PolyHighResolution.

  --cr-cutoff (float64) = Threshold for call rate. Default is 95.0 for Human, 97.0 for Diploid or Polyploid.
    default value: '95.000000'

  --fld-cutoff (float64) = Threshold for FLD
    default value: '3.600000'

  --het-so-cutoff (float64) = Threshold for HetSO
    default value: '-0.100000'

  --het-so-otv-cutoff (float64) = Threshold for HetSO OTV. Probesets with HetSO smaller than HetSO.OTV.cut are classified as OTV.
    default value: '-0.300000'

  --hom-ro-1-cutoff (float64) = Threshold for HomRO when a probeset has 1 biallelic cluster.
    default value: '0.600000'

  --hom-ro-2-cutoff (float64) = Threshold for HomRO when a probeset has 2 biallelic clusters.
    default value: '0.300000'

  --hom-ro-3-cutoff (float64) = Threshold for HomRO when a probeset has 3 biallelic clusters.
    default value: '-0.900000'

  --hom-ro (bool) = Specifies if the HomRO metrics should be used in classification.
    default value: 'True'

  --num-minor-allele-cutoff (float64) = Threshold for number of minor alleles.
    default value: '2.000000'

  --hom-ro-hap-1-cutoff (float64) = Threshold for HomRO when a probeset has 1 haploid cluster.
    default value: '0.400000'

  --hom-ro-hap-2-cutoff (float64) = Threshold for HomRO when a probeset has 2 haploid clusters.
    default value: '0.200000'

  --hom-hap-X-cutoff (float64) = Threshold for maximum distance haploid clusters can be outside of diploid clusters in the X dimension.
    default value: '-0.200000'

  --hom-hap-Y-lower-cutoff (float64) = Threshold for maximum distance haploid clusters can be above diploid clusters in the Y dimension.
    default value: '1.000000'

  --hom-hap-Y-upper-cutoff (float64) = Threshold for maximum distance haploid clusters can be below diploid clusters in the Y dimension.
    default value: '0.200000'

  --CN0-hap-X-cutoff (float64) = Threshold for maximum distance CN0 clusters can be outside of haploid clusters in the X dimension.
    default value: '-0.200000'

  --CN0-hap-Y-cutoff (float64) = Threshold for maximum distance CN0 clusters can be above haploid clusters in the Y dimension.
    default value: '0.000000'

  --CN0-dip-X-cutoff (float64) = Threshold for maximum distance CN0 clusters can be outside of diploid clusters in the X dimension.
    default value: '0.200000'

  --CN0-dip-Y-cutoff (float64) = Threshold for maximum distance CN0 clusters can be above diploid clusters in the Y dimension.
    default value: '0.000000'

  --priority-order (vec_string8) = Priority order of probeset conversion types when performing probeset selection.
    default value: 'PolyHighResolution,NoMinorHom,MonoHighResolution,OTV,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA'

  --priority-order-multi-allele (vec_string8) = Priority order of multi-allelic probeset conversion types when performing multi-allelic probeset selection.
    default value: 'PolyHighResolution,NoMinorHom,MonoHighResolution,Hemizygous,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA'

  --pid-file (filename) = Full path name of the pid file.
    default value: 'pid.ps'

  --psct-file (filename) = Full path name of the probeset-specific classification thresholds (PSCT) file.
    default value: 'psct.txt'

  --use-supplemental (bool) = Turns on the supplemental filters.
    default value: 'False'

  --homfld-cut (float64) = HomFLD.cut
    default value: '6.500000'

  --n-minor-hom-cut (float64) = nMinorHom.cut
    default value: '0.000000'

  --hetv-maf-cut (float64) = HetvMAF.cut
    default value: '10.000000'

  --variance-class (vec_string8) = List of variance.class categories. Can be a single category or a comma separated list.
    default value: 'PolyHighResolution'

  --bb-varx-cut (float64) = BB.varX.cut
    default value: '10.000000'

  --bb-vary-cut (float64) = BB.varY.cut
    default value: '10.000000'

  --ab-varx-cut (float64) = AB.varX.cut
    default value: '10.000000'

  --ab-vary-cut (float64) = AB.varY.cut
    default value: '10.000000'

  --aa-varx-cut (float64) = AA.varX.cut
    default value: '10.000000'

  --aa-vary-cut (float64) = AA.varY.cut
    default value: '10.000000'

  --bb-varx-z-cut (float64) = BB.varX.Z.cut
    default value: '3.000000'

  --bb-vary-z-cut (float64) = BB.varY.Z.cut
    default value: '3.000000'

  --ab-varx-z-cut (float64) = AB.varX.Z.cut
    default value: '3.000000'

  --ab-vary-z-cut (float64) = AB.varY.Z.cut
    default value: '3.000000'

  --aa-varx-z-cut (float64) = AA.varX.Z.cut
    default value: '3.000000'

  --aa-vary-z-cut (float64) = AA.varY.Z.cut
    default value: '3.000000'

  --aa-vary-z-cut-enabled (bool) = AA.varY.Z.cut.enabled
    default value: 'True'

  --ab-vary-z-cut-enabled (bool) = AB.varY.Z.cut.enabled
    default value: 'True'

  --bb-vary-z-cut-enabled (bool) = BB.varY.Z.cut.enabled
    default value: 'True'

  --aa-varx-z-cut-enabled (bool) = AA.varX.Z.cut.enabled
    default value: 'True'

  --ab-varx-z-cut-enabled (bool) = AB.varX.Z.cut.enabled
    default value: 'True'

  --bb-varx-z-cut-enabled (bool) = BB.varX.Z.cut.enabled
    default value: 'True'

  --use-ssp (bool) = Outputs the SSP metrics to the performance file.
    default value: 'False'

  --hom-ro-hap-1-MTChr-cutoff (float64) = Threshold for HomRO when an MT probeset has 1 cluster.
    default value: '0.400000'

  --hom-ro-hap-2-MTChr-cutoff (float64) = Threshold for HomRO when an MT probeset has 2 clusters.
    default value: '0.200000'

  --hom-ro-hap-1-XChr-cutoff (float64) = Threshold for HomRO when an X probeset has 1 cluster.
    default value: '0.100000'

  --hom-ro-hap-2-XChr-cutoff (float64) = Threshold for HomRO when an X probeset has 2 clusters.
    default value: '0.050000'

  --homfld-XChr-cut (float64) = HomFLD.cut for X probesets.
    default value: '6.500000'

  --fld-XChr-cut (float64) = FLD.cut for X probesets.
    default value: '4.000000'

  --aaf-XChr-cut (float64) = aaf.cut for X probesets.
    default value: '0.360000'

  --het-so-XChr-cutoff (float64) = Threshold for HetSO for X probesets.
    default value: '-0.100000'

  --hom-ro-hap-1-ZChr-cutoff (float64) = Threshold for HomRO when a Z probeset has 1 cluster.
    default value: '0.100000'

  --hom-ro-hap-2-ZChr-cutoff (float64) = Threshold for HomRO when a Z probeset has 2 clusters.
    default value: '0.050000'

  --homfld-ZChr-cut (float64) = HomFLD.cut for Z probesets.
    default value: '6.500000'

  --fld-ZChr-cut (float64) = FLD.cut for Z probesets.
    default value: '4.000000'

  --aaf-ZChr-cut (float64) = aaf.cut for Z probesets.
    default value: '0.360000'

  --het-so-ZChr-cutoff (float64) = Threshold for HetSO for Z probesets.
    default value: '-0.100000'

  --homfld-YChr-cut (float64) = HomFLD.cut for Y probesets.
    default value: '6.500000'

  --min-YChr-samples-cut (int32) = For Y chromosome probesets, the minimum number of called samples of each gender needed for the Euclidean distance test between male and female genotype clusters.
    default value: '5'

  --homfld-WChr-cut (float64) = HomFLD.cut for W probesets.
    default value: '6.500000'

  --min-WChr-samples-cut (int32) = For W chromosome probesets, the minimum number of called samples of each gender needed for the Euclidean distance test between male and female genotype clusters.
    default value: '5'

  --multi-metrics-file (filename) = Full path name for the multi-allelic metrics file generated by ps-metrics.

  --hom-mma-cutoff (float64) = Threshold for homozygous MMA for multi-allelic probesets.
    default value: '10.000000'

  --fld-ma-cutoff (float64) = Threshold for FLD for multi-allelic probesets.
    default value: '5.200000'

  --fld-ma-2-cutoff (float64) = Threshold for FLD for multi-allelic probesets with 2 alleles.
    default value: '5.200000'

  --min-fld-ma-cutoff (float64) = Threshold for minimal FLD for multi-allelic probesets.
    default value: '0.000000'

  --min-fld-ma-2-cutoff (float64) = Threshold for minimal FLD for multi-allelic probesets with 2 alleles.
    default value: '0.000000'

  --het-so-ma-2-cutoff (float64) = Threshold for HetSO for multi-allelic probesets with 2 alleles.
    default value: '-0.100000'

  --hom-ro-ma-cutoff (float64) = Threshold for HomRO for multi-allelic probesets.
    default value: '0.500000'

  --hom-ro-ma-1-cutoff (float64) = Threshold for HomRO-1 for multi-allelic probesets with 1 allele.
    default value: '1.000000'

  --hom-ro-ma-2-cutoff (float64) = Threshold for HomRO for multi-allelic probesets with 2 alleles.
    default value: '0.500000'

  --best-cr-ma-cutoff (float64) = Threshold for best CR for multi-allelic probesets.
    default value: '90.000000'

  --genotype-p-value-cutoff (float64) = Threshold for genotype p-values for biallelic and multi-allelic probesets.
    default value: '0.000001'

  --@help       = print developer help

</pre> </div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Monday March 11 2024 12:27:00 for Analysis Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
