<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Analysis Power Tools: VIGNETTES: Use of Analysis Power Tools (APT) to Analyze PharmacoScan™ Solution</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: Use of Analysis Power Tools (APT) to Analyze PharmacoScan™ Solution </h1>  </div>
</div>
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>February 2024</dd></dl>
<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p>PharmacoScan™ Solution can be analyzed using Axiom Analysis Suite software for Windows.  PharmacoScan can also be analyzed using scripts for Unix variants or Windows without interacting with a graphical user interface, by using Analysis Power Tools (APT).</p>
<p>The probesets in the variable copy number regions on PharmacoScan use the Copy Number Aware Genotyping (CNAG) algorithm for their correct genotyping.  The CNAG algorithm uses copy number calls to guide genotyping and reports double deletion, haploid and diploid calls.</p>
<p>This vignette demonstrates the sequence of APT steps in Linux that should be run on PharmacoScan (96 well format) for the Best Practices copy number-aware genotyping workflow.</p>
<p>The example settings and source files in this vignette assume use of the PharmacoScan_96F.r9 library package with APT version 2.12.x.  The workflow steps are the same for PharmacoScan_24F and Axiom™ PharmacoFocus™, with only minor changes needed to point to the source library files.</p>
<p><a class="anchor" id="stepstoperform"></a> </p>
<h2><a class="anchor" id="stepstoperform"></a>

<h3>Step 1. Generate signature SNP genotypes</h3>
<p>This step is optional.  It is used to make a small set of genotype calls for each sample in order to then use an unavailable tool to compare these calls against kit control reference genotypes.  The purpose is to use the genotype signature to identify the kit control samples.  Kit control samples are used to improve copy number calling.  If you already know which samples are the kit controls, you can skip this step.</p>
<p>$AXIOM_LIB_PATH is the path to the PharmacoScan_96F.r9 library folder.</p>
<p>$ORIGINAL_CEL_FILES is the path to a text file with the header “cel_files” and each subsequent row a path to each CEL file to process.</p>
<p>$OUTDIR is the path to the save location for all output for this vignette.</p>
<div class="fragment"><pre class="fragment">
apt-genotype-axiom \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/PharmacoScan_96F.r9.apt-probeset-genotype.AxiomSS1.apt2.xml \
--dual-channel-normalization true \
--cel-files $ORIGINAL_CEL_FILES \
--out-dir $OUTDIR/QC \
--log-file $OUTDIR/Logs/AxiomSS1.log
</pre></div>

<h3>Step 2. Remove samples with low “DQC” values</h3>
<p>Before performing genotyping analysis on any samples, the quality of each individual sample should be determined.  Outlier samples can be identified using Dish QC (DQC) and sample QC call rate.  DQC values are produced by the program apt-geno-qc</p>
<div class="fragment"><pre class="fragment">
apt-geno-qc \
--analysis-files-path $AXIOM_LIB_PATH \
--xml-file $AXIOM_LIB_PATH/PharmacoScan_96F.r9.apt-geno-qc.AxiomQC1.xml \
--cel-files $ORIGINAL_CEL_FILES \
--out-dir $OUTDIR/QC \
--out-file $OUTDIR/QC/Geno-qcResults.txt \
--log-file $OUTDIR/Logs/Geno-qcResults.log
</pre></div><p>
<p>The generated report file apt-geno-qc.report.txt contains QC metrics to measure the data quality of each CEL file.  Samples with a DQC value less than the DQC threshold of 0.88 should be excluded from all later steps.</p>

<h3>Step 3. Remove samples with low sample QC call rate</h3>
<p>To identify samples with low QC call rate, run genotyping using the program apt-genotype-axiom on probesets in "Step1" probeset list, which is a small list of previously wet-lab tested working probesets.</p>
<p>$CEL_LIST_INLIERS_1 is the path to a text file with the header "cel_files" and each subsequent row a path to each CEL file that meets the DQC threshold from the previous step.</p>
<div class="fragment"><pre class="fragment">
apt-genotype-axiom \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/PharmacoScan_96F_SNPSpecificPriors_Step1.r9.apt-genotype-axiom.AxiomGT1.apt2.xml \
--analysis-name GenotypingQC \
--dual-channel-normalization true \
--cel-files $CEL_LIST_INLIERS_1 \
--table-output false \
--out-dir $OUTDIR/QC \
--log-file $OUTDIR/Logs/GenotypingQC.log
</pre></div>
<p>Samples with a call rate less than 0.98 (98%) should be excluded from all later steps.  The per-sample call_rate is reported out in the GenotypingQC.report.txt file.</p>

<h3>Step 4. Generate summary signals for all probesets</h3>
<p>After excluding all samples with low QC call rate from Step 3, re-run the genotyping engine with different settings so that only summary signals will be created.  Summary signals are an input in Step 6.</p>
<p>$CEL_LIST_INLIERS_2 is the path to a text file with the header “cel_files” and each subsequent row a path to each CEL file that passes the DQC threshold AND the QC call_rate threshold from earlier steps.</p>
<div class="fragment"><pre class="fragment">
apt-genotype-axiom \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/PharmacoScan_96F.r9.apt-genotype-axiom.AxiomCN_PS1.apt2.xml \
--analysis-name ProbeSetSummarisation \
--cel-files $CEL_LIST_INLIERS_2 \
--out-dir $OUTDIR/CNData \
--log-file $OUTDIR/Logs/ProbeSetSummarisation.log
</pre></div>

<h3>Step 5. Make a list of passing kit control samples</h3>
<p>After excluding kit control samples that have low DQC or low QC call rate, make a single column text file listing the remaining kit control cel filenames.  The first row must be ‘control_samples’.  The file is normally saved to $OUTDIR\CNData\CNcontrolSamples.pass.txt and is an input in Step 6.  The file can list kit control samples spanning multiple plates in the analysis batch.  The name of the kit controls or well position does not need to be part of the filename, and the controls can be in any well.  An example file is shown below:</p>
<div class="fragment"><pre class="fragment">
control_samples
Control1_G12.CEL
Control2_H12.CEL
</pre></div>
<p>If no kit control samples pass QC, do not make this file.</p>

<h3>Step 6. Run copy number analysis</h3>
<p>apt-copynumber-axiom-cnvmix runs the fixed region copy number analysis pipeline.</p>
<div class="fragment"><pre class="fragment">
apt-copynumber-axiom-cnvmix \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/PharmacoScan_96F.r9.apt-copynumber-axiom-cnvmix.AxiomCNVmix.apt2.xml \
--reference-file $AXIOM_LIB_PATH/PharmacoScan_96F.r9.cn_models \
--summary-file $OUTDIR/CNData/ProbeSetSummarisation.summary.a5 \
--report-file $OUTDIR/CNData/ProbeSetSummarisation.report.txt \
--controls-file $OUTDIR/CNData/CNcontrolSamples.pass.txt \
--out-dir $OUTDIR/CNData \
--log-file $OUTDIR/Logs/apt-copynumber-axiom.log
</pre></div>
<p>Notes on copy number analysis step:
<ul>
<li>If you did not create CNcontrolSamples.pass.txt in Step 5, omit the controls-file parameter.</li>
<li>reference-file must be supplied with a full path, after arg-file.</li>
<li>You may delete the large input ProbeSetSummarisation.summary.a5 file once entire workflow successfully completes.</li>
<li>If you want to also do copy number analysis for SULT1A1 region, it must be done with a different arg-file (PharmacoScan_96F.r9.and_SULT1A1.apt-copynumber-axiom-cnvmix.AxiomCNVmix.apt2.xml) and omit the controls-file parameter.  This is because existing kit controls are not copy neutral for SULT1A1, so correcting SULT1A1 signals using these controls will lead to miscalling samples for SULT1A1.  Note that if you omit controls-file, copy number results for other regions may not be as good.</li>
<li>Inside the arg-file are copy number QC thresholds.  If no control samples pass both mapd-max and waviness-sd-max, then a sample must pass both thresholds to report copy number results.  If at least one control sample in controls-file passes these thresholds, then signal correction is performed using passing controls, and a sample needs to pass mapdc-max and waviness-sdc-max to report copy number results.</li>
</ul>
</p>

<h3>Step 7. Final genotyping</h3>
<p>This step will use the copy number calls from the previous step to guide genotyping.</p>
<div class="fragment"><pre class="fragment">
apt-genotype-axiom \
--copynumber-probeset-calls-file $OUTDIR/CNData/AxiomCNVMix.cnpscalls.txt \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/PharmacoScan_96F_SNPSpecificPriors_Step2.r9.apt-genotype-axiom.AxiomGT1.apt2.xml \
--dual-channel-normalization true \
--cel-files $CEL_LIST_INLIERS_2 \
--out-dir $OUTDIR \
--batch-folder $OUTDIR \
--genotyping-node:snp-posteriors-output true \
--allele-summaries true \
--log-file $OUTDIR/Logs/AxiomGT1.log
</pre></div>
<p>Notes on final genotyping step:
<ul>
<li>Not using --write-models, because it overwrites biallelic posteriors with temporary file from multiallelic genotyping.</li>
<li>Not using --multi-genotyping-node:multi-posteriors-output true, because it is already specified in arg-file.</li>
<li>Can set --allele-summaries false and omit creation of AxiomGT1.summary.txt, if you don't need this file.</li>
</ul>
</p>

<h3>Step 8. Generate SNP QC metrics</h3>
<p>This step will generate SNP QC metrics for each probeset.  Example below uses --batch-folder input instead of slower --call-file and --summary-file inputs.</p>
<div class="fragment"><pre class="fragment">
ps-metrics \
--posterior-file $OUTDIR/AxiomGT1.snp-posteriors.txt \
--multi-posterior-file $OUTDIR/AxiomGT1.snp-posteriors.multi.txt \
--batch-folder $OUTDIR \
--report-file $OUTDIR/AxiomGT1.report.txt \
--special-snps $AXIOM_LIB_PATH/PharmacoScan_96F.r9.specialSNPs \
--genotype-freq-file $AXIOM_LIB_PATH/PharmacoScan_96F.r9.genotype_frequency.txt \
--use-multi-allele true \
--y-restrict 0.2 \
--min-genotype-freq-samples 20 \
--metrics-file $OUTDIR/SNPolisher/metrics.txt \
--multi-metrics-file $OUTDIR/SNPolisher/metrics.multi.txt \
--log-file $OUTDIR/Logs/ps_metrics.log
</pre></div>

<h3>Step 9. Run SNP classification</h3>
<p>This step will read the SNP QC metrics from previous step and classify each probeset into different classes based on SNP QC metrics.</p>
<div class="fragment"><pre class="fragment">
ps-classification \
--metrics-file $OUTDIR/SNPolisher/metrics.txt \
--multi-metrics-file $OUTDIR/SNPolisher/metrics.multi.txt \
--ps2snp-file $AXIOM_LIB_PATH/PharmacoScan_96F.r9.ps2multisnp_map.ps \
--species-type Diploid \
--cr-cutoff 95 \
--fld-cutoff 3.6 \
--het-so-cutoff -0.2 \
--het-so-XChr-cutoff -0.2 \
--het-so-otv-cutoff -0.3 \
--hom-ro-1-cutoff -0.5 \
--hom-ro-2-cutoff -1.5 \
--hom-ro-3-cutoff -1.5 \
--hom-ro true \
--num-minor-allele-cutoff 2 \
--hom-ro-hap-1-cutoff -0.2 \
--hom-ro-hap-1-XChr-cutoff -0.2 \
--hom-ro-hap-1-MTChr-cutoff -0.2 \
--hom-ro-hap-2-cutoff -1.0 \
--hom-ro-hap-2-XChr-cutoff -1.0 \
--hom-ro-hap-2-MTChr-cutoff -1.0 \
--hom-hap-X-cutoff -1 \
--hom-hap-Y-lower-cutoff 1 \
--hom-hap-Y-upper-cutoff 1 \
--CN0-hap-X-cutoff -1 \
--CN0-hap-Y-cutoff -1 \
--CN0-dip-X-cutoff -1 \
--CN0-dip-Y-cutoff -1 \
--aaf-XChr-cut 1.0 \
--fld-XChr-cut 3.6 \
--homfld-XChr-cut 5.5 \
--homfld-YChr-cut 5.5 \
--min-YChr-samples-cut 5 \
--priority-order PolyHighResolution,NoMinorHom,MonoHighResolution,OTV,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA \
--recommended PolyHighResolution,NoMinorHom,MonoHighResolution,Hemizygous \
--genotype-p-value-cutoff 0.01 \
--hom-mma-cutoff 8.5 \
--fld-ma-cutoff 4.2 \
--fld-ma-2-cutoff 4.2 \
--min-fld-ma-cutoff 3.2 \
--min-fld-ma-2-cutoff 3.2 \
--het-so-ma-2-cutoff -0.2 \
--hom-ro-ma-cutoff 0.2 \
--hom-ro-ma-2-cutoff 0.3 \
--hom-ro-ma-1-cutoff 0.3 \
--priority-order-multi-allele PolyHighResolution,NoMinorHom,MonoHighResolution,Hemizygous,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA \
--best-cr-ma-cutoff 90.0 \
--use-multi-allele true \
--output-dir $OUTDIR/SNPolisher \
--log-file $OUTDIR/Logs/ps_classification.log
</pre></div>

<h3>Step 10 Perform Allele Translation</h3>
<p>Optional step.  Additional reporting is supported for a subset of markers.  The additional reports convert individual marker calls into Ref/Var call nomenclature, report haplotype pair names for some genes, and translate the haplotype pairs to predicted enzyme phenotypes.</p>
<div class="fragment"><pre class="fragment">
BASENAME=myresults
apt2-pgx-translation \
--batch-folder $OUTDIR \
--cn-region-calls-file $OUTDIR/CNData/AxiomCNVMix.cnregioncalls.txt \
--marker-list-file $OUTDIR/SNPolisher/Recommended.ps \
--analysis-files-path $AXIOM_LIB_PATH \
--translate-file PharmacoScan_96F.r9.na36.translation \
--metabolizer-file PharmacoScan_96F.r9.metabolizer \
--annotation-file PharmacoScan_96F.r9.dc_annot.csv \
--use-first-dup-allele-def true \
--out-dir $OUTDIR/translation \
--base-name-prefix $BASENAME \
--log-file $OUTDIR/Translation/$BASENAME.log
</pre></div>
<p>To perform allele translation on a subset of samples in the batch, supply the optional --sample-filter-file argument, pointing to the file containing a list of cel filenames (no path).  The first row must be ‘cel_files’.  An example file is shown below:</p>
<div class="fragment"><pre class="fragment">
cel_files
Control1_G12.CEL
Control2_H12.CEL
</pre></div>

<h3>Step 11A Make VCF using rsID as identifier</h3>
<p>Optional step.  Although /genotype/AxiomGT1.calls.txt has one copy of the original calls in one format, apt-format-result can be used to convert the calls into another format.  A common probeset filter is to restrict reporting to probesets in /SNPolisher/Recommended.ps file.  apt-format-result can use batch-folder (folder containing "AxiomAnalysisSuiteData") as input.  The following example converts calls to VCF format for recommended probesets, sets ID from the default probeset_id to the custom_rsid, and includes copy number calls.  Note that since the VCF ID must have a value, probesets with no custom_rsid will not be written to the VCF.</p>
<div class="fragment"><pre class="fragment">
BASENAME=myresults
apt-format-result \
--batch-folder $OUTDIR \
--cn-region-calls-file $OUTDIR/CNData/AxiomCNVMix.cnregioncalls.txt \
--snp-list-file $OUTDIR/SNPolisher\Recommended.ps \
--annotation-file $AXIOM_LIB_PATH/PharmacoScan_96F.na36.r9.a4.annot.db \
--snp-identifier-column custom_rsid \
--export-chr-shortname true \
--export-vcf-file $OUTDIR/Export/$BASENAME.vcf \
--log-file $OUTDIR/Export/$BASENAME.log
</pre></div>
<p>See apt-format-result –help for additional options, like making one file per sample, and selecting a subset of samples to export.</p>

<h3>Step 11B Make VCF for Axiom HLA Analysis Software</h3>
<p>Optional step.  The following parameters generate a VCF file compatible for input to Axiom HLA Analysis Software:</p>
<div class="fragment"><pre class="fragment">
BASENAME=myresults_forHLA
apt-format-result \
--batch-folder $OUTDIR \
--snp-list-file $OUTDIR/SNPolisher/Recommended.ps \
--annotation-file $AXIOM_LIB_PATH/PharmacoScan_96F.na36.r9.a4.annot.db \
--snp-identifier-column Affy_SNP_ID \
--export-chr-shortname true \
--export-vcf-file $OUTDIR/Export/$BASENAME.vcf \
--log-file $OUTDIR/export/$BASENAME.log
</pre></div>

<pre>Analysis Power Tools (APT) Release 2.12.0</pre>

</div>
<hr class="footer"/><address class="footer"><small>Generated on Wed Feb 14 2024 13:28:00 for Analysis Power Tools</small></address>
</body>
</html>
