<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Analysis Power Tools: VIGNETTES: Use of Analysis Power Tools (APT) to Analyze Axiom™ Precision Medicine Diversity Array (PMDA)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: Use of Analysis Power Tools (APT) to Analyze Axiom™ Precision Medicine Diversity Array (PMDA) </h1>  </div>
</div><!--header-->
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>2023-02-15</dd></dl>
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#stepstoperform">Best practices genotyping workflow</a></li>
<li><a href="#vcf">Save genotypes in VCF format</a></li>
<li><a href="#makexml">Make PlateNorm.apt2.xml</a></li>
</ul>

<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p>Axiom™ PMD Array can be analyzed using Axiom Analysis Suite software for Windows. Axiom™ PMD Array can also be analyzed using scripts for Unix variants or Windows without 
interacting with a graphical user interface, by using Analysis Power Tools (APT) engines for Linux and Windows.</p>
<p>The probesets in the variable copy number regions on the Axiom™ PMD Array use the Copy Number Aware Genotyping (CNAG) algorithm for their correct genotyping. The CNAG algorithm uses copy number calls to guide genotyping and reports double deletion, haploid and diploid calls. </p>
<p>This vignette demonstrates the sequence of APT steps in Linux that should be run on Axiom PMD Array for the Best Practices copy number-aware genotyping workflow with optional multiplate normalization. </p>
<p>The example settings and source files in this vignette assume use of the Axiom_PMDA.r7 library package with APT version 2.11.8. The workflow steps are the same for Axiom_PMDA_Plus.r7.1, with only minor changes needed to point to the source library files. </p>

<br />
<h2><a class="anchor" id="stepstoperform"></a>
Best practices genotyping workflow</h2>

<h3>Step 1. Generate the sample Dish QC values</h3>
<p>Before performing genotyping analysis on any samples, the quality of each individual sample should be determined. Outlier samples can be identified using Dish QC (DQC) and sample QC call rate. DQC values are produced by the program <b>apt-geno-qc</b>. </p>
<li><code>AXIOM_LIB_PATH</code> is the path to the Axiom_PMDA.r7 library folder.</li>
<li><code>ORIGINAL_CEL_FILES</code> is the path to a text file with the header “cel_files” and each subsequent row a path to each CEL file to process.</li>
<li><code>OUTDIR</code> is the path to the save location for all output for this vignette.</li>
</p>
<div class="fragment">
<pre class="fragment">
apt-geno-qc-axiom \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/Axiom_PMDA.r7.apt-geno-qc.AxiomQC1.xml \
--cel-files $ORIGINAL_CEL_FILES \
--out-file $OUTDIR/QC/apt-geno-qc.report.txt \
--log-file $OUTDIR/QC/apt-geno-qc.log
</pre>
</div>
<p>The generated report file <i>apt-geno-qc.report.txt</i> contains QC metrics to measure the data quality of each CEL file. Samples with a DQC value less than the DQC threshold of 0.82 should be excluded from next steps. </p>

<br />
<h3>Step 2. Remove samples with low sample QC call rate</h3>
<p>To identify samples with low QC call rate, run genotyping using the program <b>apt-genotype-axiom</b> on probesets in "Step1" probeset list, which is a small list of previously wet-lab tested working probesets. </p>
<li><code>CEL_LIST_INLIERS_1</code> is the path to a text file with the header "cel_files" and each subsequent row a path to each CEL file that meets the DQC threshold from the previous step. </li>
<div class="fragment">
<pre class="fragment">
apt-genotype-axiom \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/Axiom_PMDA_96orMore_Step1.r7.apt-genotype-axiom.SnpSpecificPriors.AxiomGT1.apt2.xml \
--cel-files $CEL_LIST_INLIERS_1 \
--table-output false \
--out-dir $OUTDIR/QC \
--log-file $OUTDIR/QC/apt-genotype-axiom.log
</pre>
</div><p>
<p>Notes on the QC genotyping step:
<ul>
<li>As of APT 2.11.3, dual-channel-normalization parameter is true by default, and so does not need to be specified.</li>
</ul>
</p>
<p>Samples with a call rate less than 0.97 (97%) should be excluded from the next step. The per-sample call_rate is reported out in the <i>AxiomGT1.report.txt</i> file.</p>

<br />
<h3>Step 3. Generate summary signals for all probesets</h3>
<p>After excluding all samples with low QC call rate from Step2, re-run the genotyping engine on all probesets. Note that only summary signals will be created. Summary signals are used as input for the later copy number analysis step. </p>
<li><code>CEL_LIST_INLIERS_2</code> is the path to a text file with the header “cel_files” and each subsequent row a path to each CEL file, which should all pass the DQC threshold and the QC call_rate threshold from the previous two steps. </li>
<div class="fragment">
<pre class="fragment">
apt-genotype-axiom \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/Axiom_PMDA.r7.apt-genotype-axiom.AxiomCN_PS1.apt2.xml \
--cel-files $CEL_LIST_INLIERS_2 \
--out-dir $OUTDIR/CN_input \
--log-file $OUTDIR/CN_input/apt-genotype-axiom.log
</pre>
</div>

<br />
<h3>Step 4. Run the copy number analysis</h3>
<p><b>apt-copynumber-axiom-cnvmix</b> does fixed region copy number analysis. If a sample does not pass both <code>mapd-max</code> and <code>waviness-sd-max</code>, no copy number results are reported for that sample.</p>
<div class="fragment">
<pre class="fragment">
apt-copynumber-axiom-cnvmix \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/Axiom_PMDA.r7.apt-copynumber-axiom-cnvmix.AxiomCNVmix.apt2.xml \
--reference-file $AXIOM_LIB_PATH/Axiom_PMDA.r7.cn_models \
--mapd-max 0.35 \
--waviness-sd-max 0.1 \
--summary-file $OUTDIR/CN_input/AxiomGT1.summary.a5 \
--report-file $OUTDIR/CN_input/AxiomGT1.report.txt \
--out-dir $OUTDIR/CN \
--log-file $OUTDIR/CN/apt-copynumber-axiom.log
</pre>
</div>
<p>Notes on copy number analysis step:</p>
<ul>
<li><code>reference-file</code> must be supplied with a full path, after <code>arg-file</code>.</li>
<li>Can delete the input <i>AxiomGT1.summary.a5</i> file once entire workflow successfully completes.</li>
</ul>

<br />
<h3>Step 5. Genotyping</h3>
<p>This step uses the copy number calls from the previous step to guide genotyping.</p>
<div class="fragment">
<pre class="fragment">
apt-genotype-axiom \
--copynumber-probeset-calls-file $OUTDIR/CN/AxiomCNVMix.cnpscalls.txt \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/Axiom_PMDA_96orMore_Step2.r7.apt-genotype-axiom.mm.SnpSpecificPriors.AxiomGT1.apt2.xml \
--cel-files $CEL_LIST_INLIERS_2 \
--out-dir $OUTDIR/genotypes \
--batch-folder $OUTDIR/genotypes \
--genotyping-node:snp-posteriors-output true \
--multi-genotyping-node:multi-posteriors-output true \
--allele-summaries true \
--summary-a5-output true \
--artifact-reduction-output-trustcheck true \
--do-rare-het-adjustment true \
--max-rare-het-count 3 \
--gc-channel-sci-threshold 0.39 \
--at-channel-sci-threshold 0.39 \
--major-hom-relative-min-percent 80 \
--major-hom-absolute-min-thresh 15 \
--@low-sdDist-minor-cutoff-1 1.5 \
--@high-sdDist-minor-cutoff-1 250 \
--@low-sdDist-major-cutoff-1 -9.25 \
--@high-sdDist-major-cutoff-1 0 \
--@low-sdDist-minor-cutoff-2 1.5 \
--@high-sdDist-minor-cutoff-2 250 \
--@low-sdDist-major-cutoff-2 -9.25 \
--@high-sdDist-major-cutoff-2 0 \
--@low-sdDist-minor-cutoff-3 4 \
--@high-sdDist-minor-cutoff-3 32.5 \
--@low-sdDist-major-cutoff-3 -4 \
--@high-sdDist-major-cutoff-3 -0.5 \
--@low-sdDist-minor-cutoff-4 -10 \
--@high-sdDist-minor-cutoff-4 5000 \
--@low-sdDist-major-cutoff-4 -100 \
--@high-sdDist-major-cutoff-4 5000 \
--log-file $OUTDIR/genotypes/apt-genotype-axiom.log
</pre>
</div>
<p>Notes on genotyping step:</p>
<ul>
<li>Not using <code>--write-models</code>, because it overwrites biallelic posteriors with temporary file from multiallelic genotyping.</li>
<li>As of APT 2.11.8, multiplate normalization in later steps uses the summaries file and is more efficient with HDF5 format, so <code>--allele-summaries true</code> and <code>--summary-a5-output true</code>. </li>
<li>The <i>AxiomGT1.trustcheck.txt</i> file is useful for later multiplate normalization steps to adapt to array artifacts. Since the file called by <code>arg-file</code> in this example includes <code>artifact-reduction-trustcheck true</code>, also setting <code>--artifact-reduction-output-trustcheck true</code>. </li>
<li>As of APT 2.11.3, <code>do-rare-het-adjustment</code> and associated later parameters are supported. If <code>do-rare-het-adjustment</code> is true, then additional QC tests are applied to heterozygous calls for some probesets where the count of initial heterozygous calls is between 1 and <code>max-rare-het-count</code>. If a tested heterozygous call does not pass QC, then it will be changed to NoCall. The setting <code>do-rare-het-adjustment true</code> is recommended for human arrays for analysis batches of at least 90 samples. Since Axiom_PMDA.r7 library package doesn’t include the <code>do-rare-het-adjustment</code> and associated parameters, they have been added to this analysis step. </li>
<li>As of APT 2.11.3, dual-channel-normalization parameter is true by default, and so does not need to be specified.</li>
</ul>

<br />
<h3>Step 6. Generate SNP QC metrics</h3>
<p>This step generates SNP QC metrics for each probeset. Example below uses <code>--batch-folder</code> input instead of slower <code>--call-file</code> and <code>--summary-file</code> inputs. </p>
<div class="fragment">
<pre class="fragment">
ps-metrics \
--posterior-file $OUTDIR/genotypes/AxiomGT1.snp-posteriors.txt \
--multi-posterior-file $OUTDIR/genotypes/AxiomGT1.snp-posteriors.multi.txt \
--batch-folder $OUTDIR/genotypes \
--report-file $OUTDIR/genotypes/AxiomGT1.report.txt \
--special-snps $AXIOM_LIB_PATH/Axiom_PMDA.r7.specialSNPs \
--use-multi-allele true \
--y-restrict 0.2 \
--min-genotype-freq-samples 20 \
--metrics-file $OUTDIR/SNPolisher/metrics.txt \
--multi-metrics-file $OUTDIR/SNPolisher/metrics.multi.txt \
--log-file $OUTDIR/SNPolisher/ps_metrics.log
</pre>
</div>

<br />
<h3>Step 7. Run SNP classification</h3>
<p>This step uses the SNP QC metrics from the previous step to classify each probeset. </p>
<div class="fragment">
<pre class="fragment">
ps-classification \
--metrics-file $OUTDIR/SNPolisher/metrics.txt \
--multi-metrics-file $OUTDIR/SNPolisher/metrics.multi.txt \
--ps2snp-file $AXIOM_LIB_PATH/Axiom_PMDA.r7.ps2multisnp_map.ps \
--species-type Human \
--cr-cutoff 95 \
--fld-cutoff 3.6 \
--het-so-cutoff -0.1 \
--het-so-XChr-cutoff -0.1 \
--het-so-otv-cutoff -0.3 \
--hom-ro-1-cutoff 0.6 \
--hom-ro-2-cutoff 0.3 \
--hom-ro-3-cutoff -0.9 \
--hom-ro true \
--num-minor-allele-cutoff 2 \
--hom-ro-hap-1-cutoff 0.1 \
--hom-ro-hap-1-XChr-cutoff 0.1 \
--hom-ro-hap-1-MTChr-cutoff 0.4 \
--hom-ro-hap-2-cutoff -0.9 \
--hom-ro-hap-2-XChr-cutoff 0.05 \
--hom-ro-hap-2-MTChr-cutoff 0.2 \
--hom-hap-X-cutoff -1 \
--hom-hap-Y-lower-cutoff 1 \
--hom-hap-Y-upper-cutoff 1 \
--CN0-hap-X-cutoff -1 \
--CN0-hap-Y-cutoff -1 \
--CN0-dip-X-cutoff -1 \
--CN0-dip-Y-cutoff -1 \
--aaf-XChr-cut 0.36 \
--fld-XChr-cut 4 \
--homfld-XChr-cut 6.5 \
--homfld-YChr-cut 6.5 \
--min-YChr-samples-cut 5 \
--priority-order PolyHighResolution,NoMinorHom,MonoHighResolution,OTV,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA \
--recommended PolyHighResolution,NoMinorHom,MonoHighResolution,Hemizygous \
--genotype-p-value-cutoff 0.000001 \
--hom-mma-cutoff 8.5 \
--fld-ma-cutoff 4.2 \
--fld-ma-2-cutoff 4.2 \
--min-fld-ma-cutoff 3.2 \
--min-fld-ma-2-cutoff 3.2 \
--het-so-ma-2-cutoff -0.2 \
--hom-ro-ma-cutoff 0.2 \
--hom-ro-ma-2-cutoff 0.3 \
--hom-ro-ma-1-cutoff 0.3 \
--priority-order-multi-allele PolyHighResolution,NoMinorHom,MonoHighResolution,Hemizygous,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA \
--best-cr-ma-cutoff 90.0 \
--use-multi-allele true \
--output-dir $OUTDIR/SNPolisher \
--log-file $OUTDIR/SNPolisher/ps_classification.log
</pre>
</div>

<br />
<h3>Step 8. MPN signal summarization</h3>
<p>Multiplate normalization (MPN) is an optional set of steps applied after the first round of genotyping and SNP QC. Multiplate normalization corrects for between-plate effects by adjusting the signals for some probesets for entire plates. Multiplate normalization can improve genotyping call rate, accuracy, and the number of probesets that pass SNP QC. Multiplate normalization is recommended for multiple plate genotyping analysis. </p>
<p>This step first checks if the data is appropriate for multiplate normalization. Using the vignette parameters, the data must have: </p>
<li>at least two plates or two scans of one plate with at least eight samples each </li>
<li>at least one biallelic, non-rare-het adjusted probeset whose SNP QC ConversionType is one of: CallRateBelowThreshold, OTV, UnexpectedGenotypeFreq, Other. </li>
<p>If this data is not appropriate for multiplate normalization, then the workflow will stop in the next step. If this data is appropriate for multiplate normalization, then candidate probesets have their summary signals adjusted in a new <i>PlateNorm.summary.a5</i> file. </p>
<p>If multiplate normalization in later steps updates the genotypes for at least one probeset, an updated <i>Ps.performance.txt</i> file is created, overwriting the original file. The original <i>Ps.performance.txt</i> from previous step is an input for this workflow step. For traceability, the vignette makes a backup of the original <i>Ps.performance.txt</i> and uses the backup as input for this step. </p>
<p></p>
<pre class="fragment">
cp $OUTDIR/SNPolisher/Ps.performance.txt $OUTDIR/SNPolisher/Ps.performance.txt.bak
</pre>
<pre class="fragment">
apt2-summary-plate-norm \
--input-summary-file $OUTDIR/genotypes/AxiomGT1.summary.a5 \
--plate-map-file $OUTDIR/genotypes/AxiomGT1.report.txt \
--rare-het-report-file $OUTDIR/genotypes/AxiomGT1.rare_het.report.txt \
--ps-performance-file $OUTDIR/SNPolisher/Ps.performance.txt.bak \
--out-dir $OUTDIR/PlateNorm \
--analysis-name PlateNorm \
--exclude-multisnp-biallele false \
--log-file $OUTDIR/PlateNorm/apt2-summary-plate-norm.log
</pre>
<p>Notes on multiplate signal summarization: </p>
<ul>
<li><code>plate-map-file</code> is used to group samples by plate. </li>
<li><code>ps-performance-file</code>, <code>rare-het-report-file</code>, and <code>exclude-multisnp-biallele</code> are used to identify candidate probesets for per-plate signal adjustment and regenotyping. </li>
</ul>

<br />
<h3>Step 9. MPN first early stopping point</h3>
<p>This step ends the workflow now if no probesets are candidates for multiplate normalization. The original genotype results are not changed. </p>
<pre class="fragment">
file=$OUTDIR/PlateNorm/PlateNorm.norm.ps
if ! [ -f $file ]; then
    echo 'PlateNorm.norm.ps does not exist. Exit the script.' 
    exit 1
fi
</pre>

<br />
<h3><a class="anchor" id="mpngenotyping"></a>
Step 10. MPN genotyping</h3>
<p>This step genotypes the multiplate normalized probesets.</p>
<pre class="fragment">
apt-summary-genotype-axiom \
--analysis-files-path $AXIOM_LIB_PATH \
--arg-file $AXIOM_LIB_PATH/Axiom_PMDA.r7.apt-summary-genotype-axiom.PlateNorm.apt2.xml \
--analysis-name PlateNorm \
--summary-a5-input-file $OUTDIR/PlateNorm/PlateNorm.summary.a5 \
--probeset-ids $OUTDIR/PlateNorm/PlateNorm.norm.ps \
--gender-file $OUTDIR/genotypes/AxiomGT1.report.txt \
--artifact-reduction-trustcheck-file $OUTDIR/genotypes/AxiomGT1.trustcheck.txt \
--copynumber-probeset-calls-file $OUTDIR/CN/AxiomCNVMix.cnpscalls.txt \
--out-dir $OUTDIR/PlateNorm \
--batch-folder $OUTDIR/PlateNorm \
--genotyping-node:snp-posteriors-output true \
--log-file $OUTDIR/PlateNorm/apt-summary-genotype-axiom.log
</pre>
<p>Notes on the MPN genotyping step: </p>
<ul>
<li>Since the starting point is not CEL files, a different genotyping engine is used than in step 5. </li>
<li>Refer to <a href="#makexml">Make PlateNorm.apt2.xml</a> if the file used with arg-file parameter is not available. </li>
</ul>

<br />
<h3>Step 11. MPN generate SNP QC metrics</h3>
<p>SNP QC determines which regenotyped probesets are appropriate for use. This step generates SNP QC metrics for each multiplate normalized probeset. </p>
<pre class="fragment">
ps-metrics \
--posterior-file $OUTDIR/PlateNorm/PlateNorm.snp-posteriors.txt \
--batch-folder $OUTDIR/PlateNorm \
--report-file $OUTDIR/PlateNorm/PlateNorm.report.txt \
--special-snps $AXIOM_LIB_PATH/Axiom_PMDA.r7.specialSNPs \
--y-restrict 0.2 \
--min-genotype-freq-samples 20 \
--metrics-file $OUTDIR/PlateNorm/SNPolisher/metrics.txt \
--log-file $OUTDIR/PlateNorm/SNPolisher/ps_metrics.log
</pre>

<br />
<h3>Step 12. MPN SNP classification</h3>
<p>This step uses the SNP QC metrics from the previous step to classify each probeset. Compared with the step 7 classification run, this step must not include the <code>ps2snp-file</code> parameter and uses a larger threshold for both <code>fld-cutoff</code> and <code>fld-XChr-cut</code>. Since the previous step did not create a multi-metrics-file, this step does not include <code>multi-metrics-file</code> or <code>use-multi-allele</code> parameters. </p>
<pre class="fragment">
ps-classification \
--metrics-file $OUTDIR/PlateNorm/SNPolisher/metrics.txt \
--species-type Human \
--cr-cutoff 95 \
--fld-cutoff 4.9 \
--het-so-cutoff -0.1 \
--het-so-XChr-cutoff -0.1 \
--het-so-otv-cutoff -0.3 \
--hom-ro-1-cutoff 0.6 \
--hom-ro-2-cutoff 0.3 \
--hom-ro-3-cutoff -0.9 \
--hom-ro true \
--num-minor-allele-cutoff 2 \
--hom-ro-hap-1-cutoff 0.1 \
--hom-ro-hap-1-XChr-cutoff 0.1 \
--hom-ro-hap-1-MTChr-cutoff 0.4 \
--hom-ro-hap-2-cutoff -0.9 \
--hom-ro-hap-2-XChr-cutoff 0.05 \
--hom-ro-hap-2-MTChr-cutoff 0.2 \
--hom-hap-X-cutoff -1 \
--hom-hap-Y-lower-cutoff 1 \
--hom-hap-Y-upper-cutoff 1 \
--CN0-hap-X-cutoff -1 \
--CN0-hap-Y-cutoff -1 \
--CN0-dip-X-cutoff -1 \
--CN0-dip-Y-cutoff -1 \
--aaf-XChr-cut 0.36 \
--fld-XChr-cut 4.9 \
--homfld-XChr-cut 6.5 \
--homfld-YChr-cut 6.5 \
--min-YChr-samples-cut 5 \
--priority-order PolyHighResolution,NoMinorHom,MonoHighResolution,OTV,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA \
--recommended PolyHighResolution,NoMinorHom,MonoHighResolution,Hemizygous \
--genotype-p-value-cutoff 0.000001 \
--hom-mma-cutoff 8.5 \
--fld-ma-cutoff 4.2 \
--fld-ma-2-cutoff 4.2 \
--min-fld-ma-cutoff 3.2 \
--min-fld-ma-2-cutoff 3.2 \
--het-so-ma-2-cutoff -0.2 \
--hom-ro-ma-cutoff 0.2 \
--hom-ro-ma-2-cutoff 0.3 \
--hom-ro-ma-1-cutoff 0.3 \
--priority-order-multi-allele PolyHighResolution,NoMinorHom,MonoHighResolution,Hemizygous,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA \
--best-cr-ma-cutoff 90.0 \
--output-dir $OUTDIR/PlateNorm/SNPolisher \
--log-file $OUTDIR/PlateNorm/SNPolisher/ps_classification.log
</pre>

<br />
<h3>Step 13. MPN second early stopping point</h3>
<p>This step ends the workflow now if no regenotyped probesets pass SNP QC. The original genotype results are not changed. </p>
<pre class="fragment">
file=$OUTDIR/PlateNorm/SNPolisher/Recommended.ps
if ! [ -f $file ]; then
    echo 'PlateNorm Recommended.ps does not exist. Exit the script.'
    exit 1
fi
</pre>

<br />
<h3>Step 14. MPN update original genotype results</h3>
<p>This step merges the multiplate normalization genotype results for recommended probesets into the main genotype results folder. </p>
<pre class="fragment">
ps-update \
--special-snps $AXIOM_LIB_PATH/Axiom_PMDA.r7.specialSNPs \
--main-gt-dir $OUTDIR/genotypes \
--main-analysis-name AxiomGT1 \
--main-metrics-file $OUTDIR/SNPolisher/metrics.txt \
--supplemental-gt-dir $OUTDIR/PlateNorm \
--supplemental-analysis-name PlateNorm \
--supplemental-metrics-file $OUTDIR/PlateNorm/SNPolisher/metrics.txt \
--probeset-ids $OUTDIR/PlateNorm/SNPolisher/Recommended.ps \
--full-backup false \
--log-file $OUTDIR/ps-update.log
</pre>
<p>Notes on MPN update of original genotypes: </p>
<ul>
<li>Since this example does not include <code>out-gt-dir</code> or <code>out-ps-metrics-dir</code> pointing to a new folder, results in the supplemental 'PlateNorm' folders are merged into the main folder files. When <b>ps-update</b> merges results into the main folders, it makes a backup of the original results. Since <code>--full-backup false</code>, a small backup is made when possible, containing the original results for only the probesets that are later updated. </li>
<li>The 'AxiomAnalysisSuiteData' folder will not be updated unless it is a subfolder of <code>main-gt-dir</code>. </li>
<li>Refer to the log file for the list of files that are updated.</li>
</ul>

<br />
<h3>Step 15. Update SNP classification</h3>
<p>The final step of this workflow uses the updated SNP QC metrics from the previous step to do final classification of all probesets. The parameters are the same as in step 7 except for the name of the new log file. The step 7 <i>Ps.performance.txt</i> file was backed up in step 8. Additional step 7 outputs are renamed now so that <b>ps-classification</b> does not overwrite existing files. </p>
<pre class="fragment">
# Rename .ps files to .ps.bak
sources=( $OUTDIR/SNPolisher/*.ps )
for file in "${sources[@]}"; do
    mv "$file" "${file%. *}.bak"
done
</pre>
<pre class="fragment">
ps-classification \
--metrics-file $OUTDIR/SNPolisher/metrics.txt \
--multi-metrics-file $OUTDIR/SNPolisher/metrics.multi.txt \
--ps2snp-file $AXIOM_LIB_PATH/Axiom_PMDA.r7.ps2multisnp_map.ps \
--species-type Human \
--cr-cutoff 95 \
--fld-cutoff 3.6 \
--het-so-cutoff -0.1 \
--het-so-XChr-cutoff -0.1 \
--het-so-otv-cutoff -0.3 \
--hom-ro-1-cutoff 0.6 \
--hom-ro-2-cutoff 0.3 \
--hom-ro-3-cutoff -0.9 \
--hom-ro true \
--num-minor-allele-cutoff 2 \
--hom-ro-hap-1-cutoff 0.1 \
--hom-ro-hap-1-XChr-cutoff 0.1 \
--hom-ro-hap-1-MTChr-cutoff 0.4 \
--hom-ro-hap-2-cutoff -0.9 \
--hom-ro-hap-2-XChr-cutoff 0.05 \
--hom-ro-hap-2-MTChr-cutoff 0.2 \
--hom-hap-X-cutoff -1 \
--hom-hap-Y-lower-cutoff 1 \
--hom-hap-Y-upper-cutoff 1 \
--CN0-hap-X-cutoff -1 \
--CN0-hap-Y-cutoff -1 \
--CN0-dip-X-cutoff -1 \
--CN0-dip-Y-cutoff -1 \
--aaf-XChr-cut 0.36 \
--fld-XChr-cut 4 \
--homfld-XChr-cut 6.5 \
--homfld-YChr-cut 6.5 \
--min-YChr-samples-cut 5 \
--priority-order PolyHighResolution,NoMinorHom,MonoHighResolution,OTV,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA \
--recommended PolyHighResolution,NoMinorHom,MonoHighResolution,Hemizygous \
--genotype-p-value-cutoff 0.000001 \
--hom-mma-cutoff 8.5 \
--fld-ma-cutoff 4.2 \
--fld-ma-2-cutoff 4.2 \
--min-fld-ma-cutoff 3.2 \
--min-fld-ma-2-cutoff 3.2 \
--het-so-ma-2-cutoff -0.2 \
--hom-ro-ma-cutoff 0.2 \
--hom-ro-ma-2-cutoff 0.3 \
--hom-ro-ma-1-cutoff 0.3 \
--priority-order-multi-allele PolyHighResolution,NoMinorHom,MonoHighResolution,Hemizygous,UnexpectedGenotypeFreq,CallRateBelowThreshold,Other,OtherMA \
--best-cr-ma-cutoff 90.0 \
--use-multi-allele true \
--output-dir $OUTDIR/SNPolisher \
--log-file $OUTDIR/SNPolisher/ps_classification_update.log
</pre>

<br />
<h2><a class="anchor" id="vcf"></a>
Save genotypes in VCF format</h2>
<p>Although <i>/genotype/AxiomGT1.calls.txt</i> has one copy of the original calls in one format, <b>apt-format-result</b> can be used to convert the calls into another format. <b>apt-format-result</b> can filter to smaller set of samples and probesets. A common probeset filter is to restrict reporting to probesets in <i>/SNPolisher/Recommended.ps</i>. <b>apt-format-result</b> can use <code>batch-folder</code> (folder containing 'AxiomAnalysisSuiteData') as input. The following example converts calls to VCF format for recommended probesets, sets ID from the default probeset_id to the extended_rsid, and includes copy number calls. Note that since the VCF ID must have a value, probesets with no extended_rsid will not be written to the VCF. </p>
<pre class="fragment">
apt-format-result \
--batch-folder $OUTDIR/genotypes \
--cn-region-calls-file $OUTDIR/CN/AxiomCNVMix.cnregioncalls.txt \
--snp-list-file $OUTDIR/SNPolisher/Recommended.ps \
--annotation-file $AXIOM_LIB_PATH/Axiom_PMDA.na36.r7.a8.annot.db \
--snp-identifier-column extended_rsid \
--export-chr-shortname true \
--export-vcf-file $OUTDIR/export/myresults.vcf \
--log-file $OUTDIR/export/myresults.log
</pre>

<br />
<h2><a class="anchor" id="makexml"></a>
Make PlateNorm.apt2.xml</h2>
<p>These instructions are for making the <i>PlateNorm.apt2.xml</i> file used by <code>arg-file</code> parameter in <a href="#mpngenotyping">Step 10. MPN genotyping</a>, if the file is not available. </p>
<ol>
<li>Install Axiom Analysis Suite 5.3 or newer desktop application. </li>
<li>Install the analysis library package in the library folder used by Axiom Analysis Suite. </li>
<li>Start Axiom Analysis Suite. </li>
<li>Import at least two plates of at least 10 CEL files each. Verify that the correct Array Type is selected. </li>
<li>Select the Genotyping workflow. </li>
<li>In the Analysis Thresholds pane, select an analysis thresholds file if none is selected. </li>
<li>In the Analysis Settings pane:Genotyping subsection, confirm that Multiplate Normalization Enabled checkbox is checked. </li>
<li>Enter the batch name in the Name field and click Run. </li>
<li>Switch to the Dashboard tab. When the batch has completed processing, control-click on the batch row and select 'Open in Windows Explorer.' </li>
<li>Open the selected folder and browse to file <i>\Temp\apt-summary-genotype-axiom.PlateNorm.apt2.xml</i>. This file will exist if at least one probeset was a candidate for multiplate normalization in this batch. </li>
<li>Copy <i>apt-summary-genotype-axiom.PlateNorm.apt2.xml</i> to the library folder for the array, preferably with the name <i>[axiom_array].r[#].apt-summary-genotype-axiom.PlateNorm.apt2.xml</i>. For this vignette, the name would be <i>Axiom_PMDA.r7.apt-summary-genotype-axiom.PlateNorm.apt2.xml</i>. </li>
<li>The new file can now be used for <code>--arg-file</code> parameter by <b>apt-summary-genotype-axiom</b>. </li>
</ol>

</div></div>
<hr class="footer"/><address class="footer"><small>Generated on Wed Feb 15 2023 10:44:00 for Analysis Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
